---
description: >-
  This R markdown file was generated by Dr Richard Barker to provide insight
  into Arabidopsis.
---

# Reproducing APEX05 iDEP linear model analyses with R Markdown

### 1. Read data

First we set up the working directory to where the files are saved.

```
 setwd('/Users/drbhomeoffice/Creative Cloud Files/apex05_run1_gene_counts_from_Lucas_pipeline/iDEP_fix2_cax22_cax23_col_4_3_factor_statistical modeles/iDEP_fix2_analysis_cax22_cax23_col_presentation_V5_4_manuscript/IDEP_R_code_for_markdown_dev/APEX05_V10_linear_model_r-Markdown')   # Needs to be changed  
```

R packages and iDEP core Functions. .

```
 if(file.exists('iDEP_core_functions.R'))
    source('iDEP_core_functions.R') else 
    source('https://raw.githubusercontent.com/iDEP-SDSU/idep/master/shinyapps/idep/iDEP_core_functions.R') 
```

```
## Warning in install.packages(...): installation of package 'runibic' had non-zero
## exit status
```

```
## Warning: dependencies 'OpenMPController', 'randomForest', 'RandomFields',
## 'equate', 'phangorn' are not available
```

```
## 
##   There are binary versions available but the source versions are later:
##                       binary     source needs_compilation
## polspline             1.1.19     1.1.20              TRUE
## qpdf                     1.1      1.3.0              TRUE
## s2                     1.0.7      1.1.0              TRUE
## terra                 1.5-21     1.6-17              TRUE
## RMariaDB               1.2.1      1.2.2              TRUE
## RPostgres              1.4.3      1.4.4              TRUE
## RPostgreSQL            0.7-3      0.7-4              TRUE
## lobstr                 1.1.1      1.1.2              TRUE
## clue                  0.3-60     0.3-61              TRUE
## SGP                  1.9-5.0    2.0-0.0             FALSE
## taxize                0.9.99    0.9.100             FALSE
## chron                 2.3-56     2.3-58              TRUE
## rms                    6.2-0      6.3-0              TRUE
## VGAM                   1.1-6      1.1-7              TRUE
## pdftools               3.1.1      3.3.1              TRUE
## party                  1.3-9     1.3-11              TRUE
## sf                     1.0-7      1.0-8              TRUE
## PBSmapping            2.73.0     2.73.2              TRUE
## raster                3.5-15      3.6-3              TRUE
## mix                   1.0-10     1.0-11              TRUE
## sodium                 1.2.0      1.2.1              TRUE
## logspline             2.1.16     2.1.17              TRUE
## DendSer                1.0.1      1.0.2              TRUE
## umap                 0.2.8.0    0.2.9.0              TRUE
## rgdal                 1.5-29     1.5-32              TRUE
## spam64                 2.8-0      2.9-0              TRUE
## sm                   2.2-5.7  2.2-5.7.1              TRUE
## RandomFieldsUtils      1.2.3      1.2.5              TRUE
## nleqslv                3.3.2      3.3.3              TRUE
## tweedie                2.3.3      2.3.5              TRUE
## archive                1.1.4      1.1.5              TRUE
## infotheo               1.2.0    1.2.0.1              TRUE
## strucchange            1.5-2      1.5-3              TRUE
## tseries              0.10-50    0.10-52              TRUE
## AlgDesign              1.2.0      1.2.1              TRUE
## Cairo                 1.5-15      1.6-0              TRUE
## callr                  3.7.0      3.7.2             FALSE
## checkmate              2.0.0      2.1.0              TRUE
## classInt               0.4-3      0.4-8              TRUE
## cli                    3.2.0      3.4.1              TRUE
## cluster                2.1.3      2.1.4              TRUE
## conquer                1.3.0      1.3.1              TRUE
## covr                   3.5.1      3.6.1              TRUE
## curl                   4.3.2      4.3.3              TRUE
## dbplyr                 2.1.1      2.2.1             FALSE
## devtools               2.4.3      2.4.5             FALSE
## dotCall64              1.0-1      1.0-2              TRUE
## dplyr                  1.0.8     1.0.10              TRUE
## e1071                  1.7-9     1.7-11              TRUE
## farver                 2.1.0      2.1.1              TRUE
## fields                  13.3       14.1              TRUE
## flexclust              1.4-0      1.4-1              TRUE
## FNN                    1.1.3    1.1.3.1              TRUE
## foreign               0.8-82     0.8-83              TRUE
## geometry               0.4.5    0.4.6.1              TRUE
## gert                   1.6.0      1.9.1              TRUE
## gmp                    0.6-5      0.6-6              TRUE
## gtools                 3.9.2      3.9.3              TRUE
## haven                  2.4.3      2.5.1              TRUE
## hdf5r                  1.3.5      1.3.7              TRUE
## Hmisc                  4.6-0      4.7-1              TRUE
## htmltools              0.5.2      0.5.3              TRUE
## httpuv                 1.6.5      1.6.6              TRUE
## hunspell               3.0.1      3.0.2              TRUE
## igraph                 1.3.0      1.3.5              TRUE
## interp                1.0-33      1.1-3              TRUE
## ipred                 0.9-12     0.9-13              TRUE
## irlba                  2.3.5    2.3.5.1              TRUE
## isoband                0.2.5      0.2.6              TRUE
## jsonlite               1.8.0      1.8.2              TRUE
## JuliaCall             0.17.4     0.17.5              TRUE
## kernlab               0.9-30     0.9-31              TRUE
## lifecycle              1.0.1      1.0.3             FALSE
## lme4                  1.1-28     1.1-30              TRUE
## lpSolve               5.6.15     5.6.17              TRUE
## maptools               1.1-3      1.1-4              TRUE
## MASS                  7.3-56   7.3-58.1              TRUE
## Matrix                 1.4-1      1.5-1              TRUE
## MatrixModels           0.5-0      0.5-1             FALSE
## matrixStats           0.61.0     0.62.0              TRUE
## mclust                 5.4.9     5.4.10              TRUE
## memisc             0.99.30.7  0.99.31.3              TRUE
## mnormt                 2.0.2      2.1.1              TRUE
## nlme                 3.1-157    3.1-160              TRUE
## nloptr                 2.0.0      2.0.3              TRUE
## nnet                  7.3-17     7.3-18              TRUE
## openssl                2.0.0      2.0.3              TRUE
## osqp                 0.6.0.5    0.6.0.6              TRUE
## pkgload                1.2.4      1.3.0             FALSE
## processx               3.5.3      3.7.0              TRUE
## proxy                 0.4-26     0.4-27              TRUE
## ps                     1.6.0      1.7.1              TRUE
## purrr                  0.3.4      0.3.5              TRUE
## qap                    0.1-1      0.1-2              TRUE
## quantreg                5.88       5.94              TRUE
## ragg                   1.2.2      1.2.3              TRUE
## Rcpp                 1.0.8.3      1.0.9              TRUE
## RcppArmadillo     0.11.0.0.0 0.11.4.0.1              TRUE
## RcppEigen          0.3.3.9.1  0.3.3.9.2              TRUE
## RCurl               1.98-1.6   1.98-1.9              TRUE
## readr                  2.1.2      2.1.3              TRUE
## readxl                 1.4.0      1.4.1              TRUE
## reticulate              1.24       1.26              TRUE
## rgl                  0.108.3    0.110.2              TRUE
## rlang                  1.0.2      1.0.6              TRUE
## Rmpfr                  0.8-7      0.8-9              TRUE
## RNetCDF                2.5-2      2.6-1              TRUE
## roxygen2               7.1.2      7.2.1              TRUE
## RSpectra              0.16-0     0.16-1              TRUE
## RSQLite               2.2.12     2.2.18              TRUE
## Rtsne                   0.15       0.16              TRUE
## Rttf2pt1              1.3.10     1.3.11              TRUE
## sass                   0.4.1      0.4.2              TRUE
## sctransform            0.3.3      0.3.5              TRUE
## seriation              1.3.5      1.3.6              TRUE
## Seurat                 4.1.0      4.2.0              TRUE
## SeuratObject           4.0.4      4.1.2              TRUE
## sp                     1.4-6      1.5-0              TRUE
## spam                   2.8-0      2.9-1              TRUE
## spatstat.core          2.4-2      2.4-4              TRUE
## spatstat.sparse        2.1-0      2.1-1              TRUE
## spatstat.utils         2.3-0      2.3-1              TRUE
## statmod               1.4.36     1.4.37              TRUE
## stringi                1.7.6      1.7.8              TRUE
## survival               3.3-1      3.4-0              TRUE
## testthat               3.1.3      3.1.5              TRUE
## tibble                 3.1.6      3.1.8              TRUE
## tidyr                  1.2.0      1.2.1              TRUE
## tidyselect             1.1.2      1.2.0             FALSE
## TSP                    1.2-0      1.2-1              TRUE
## ucminf                 1.1-4    1.1-4.1              TRUE
## uuid                   1.0-4      1.1-0              TRUE
## uwot                  0.1.11     0.1.14              TRUE
## V8                     4.1.0      4.2.1              TRUE
## vctrs                  0.4.0      0.4.2              TRUE
## vegan                  2.5-7      2.6-4              TRUE
## vroom                  1.5.7      1.6.0              TRUE
## webfakes               1.1.3      1.1.4              TRUE
## WGCNA                 1.70-3       1.71              TRUE
## xfun                    0.30       0.33              TRUE
## XML                 3.99-0.9  3.99-0.11              TRUE
## zip                    2.2.0      2.2.1              TRUE
## zoo                    1.8-9     1.8-11              TRUE
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'polspline' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'RPostgreSQL' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'chron' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'VGAM' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'PBSmapping' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'mix' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'sodium' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'logspline' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'spam64' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'sm' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'RandomFieldsUtils' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'nleqslv' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'tweedie' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'infotheo' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'AlgDesign' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'Cairo' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'checkmate' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'cli' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'cluster' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'curl' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'dotCall64' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'farver' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'flexclust' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'FNN' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'foreign' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'gmp' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'gtools' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'hdf5r' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'isoband' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'jsonlite' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'kernlab' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'lpSolve' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'MASS' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'Matrix' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'matrixStats' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'mclust' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'mnormt' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'nlme' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'nnet' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'openssl' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'proxy' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'ps' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'qap' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'ragg' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'Rcpp' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'RCurl' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'rlang' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'RNetCDF' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'Rttf2pt1' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'sp' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'spatstat.utils' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'statmod' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'stringi' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'TSP' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'ucminf' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'uuid' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'webfakes' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'xfun' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'XML' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'zip' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'zoo' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'qpdf' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 's2' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'terra' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'RMariaDB' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'RPostgres' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'lobstr' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'clue' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'rgdal' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'strucchange' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'tseries' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'covr' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'e1071' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'geometry' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'gert' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'htmltools' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'httpuv' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'hunspell' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'igraph' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'irlba' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'JuliaCall' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'lifecycle' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'maptools' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'MatrixModels' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'memisc' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'osqp' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'pkgload' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'processx' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'purrr' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'RcppArmadillo' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'RcppEigen' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'reticulate' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'Rmpfr' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'RSQLite' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'Rtsne' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'seriation' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'spam' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'spatstat.sparse' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'survival' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'V8' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'vctrs' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'vegan' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'pdftools' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'party' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'raster' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'DendSer' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'callr' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'classInt' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'conquer' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'fields' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'Hmisc' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'interp' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'ipred' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'quantreg' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'rgl' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'roxygen2' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'RSpectra' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'sass' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'SeuratObject' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'spatstat.core' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'tibble' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'tidyselect' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'uwot' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'SGP' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'taxize' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'rms' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'sf' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'umap' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'archive' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'dplyr' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'readxl' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'testthat' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'vroom' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'WGCNA' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'dbplyr' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'devtools' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'nloptr' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'readr' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'sctransform' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'tidyr' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'haven' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'lme4' had non-zero exit status
```

```
## Warning in install.packages(update[instlib == l, "Package"], l, repos = repos, :
## installation of package 'Seurat' had non-zero exit status
```

```
## Warning: package 'dplyr' was built under R version 4.0.5
```

```
## Warning: package 'Rcpp' was built under R version 4.0.5
```

```
## Warning: package 'MASS' was built under R version 4.0.5
```

```
## Warning: package 'colorspace' was built under R version 4.0.5
```

```
## Warning: package 'igraph' was built under R version 4.0.5
```

```
## Warning: Package 'KEGG.db' is deprecated and will be removed from Bioconductor
##   version 3.12
```

```
## Warning: package 'XML' was built under R version 4.0.5
```

```
## Warning in library(package, lib.loc = lib.loc, character.only = TRUE,
## logical.return = TRUE, : there is no package called 'runibic'
```

```
## Warning: package 'sfsmisc' was built under R version 4.0.5
```

```
## 
## 
## Warnning!!!!!! These Bioconductor packages cannot be loaded: runibic STRINGdb
```

We are using the downloaded gene expression file where gene IDs has been converted to Ensembl gene IDs.

```
 inputFile <- 'Downloaded_Converted_Data.csv'  # Expression matrix
 sampleInfoFile <- 'run1_gene_counts_factors_v2.2_col_cax22_cax23_row_removed.csv' # Experiment design file 
 geneInfoFile <- 'Arabidopsis_thaliana__athaliana_eg_gene_GeneInfo.csv' #Gene symbols, location etc. 
 geneSetFile <- 'Arabidopsis thaliana__athaliana_eg_gene.db'  # pathway database in SQL; can be GMT format 
 STRING10_speciesFile <- 'https://raw.githubusercontent.com/iDEP-SDSU/idep/master/shinyapps/idep/STRING10_species.csv' 
```

Parameters for reading data

```
 input_missingValue <- 'geneMedianInGroup'  #Missing values imputation method
 input_dataFileFormat <- 1  #1- read counts, 2 FKPM/RPKM or DNA microarray
 input_minCounts <- 0.5 #Min counts
 input_NminSamples <- 1 #Minimum number of samples 
 input_countsLogStart <- 4  #Pseudo count for log CPM
 input_CountsTransform <- 3 #Methods for data transformation of counts. 1-EdgeR's logCPM 2-VST, 3-rlog 
```

```
 readData.out <- readData(inputFile) 
```

```
## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors
```

```
 library(knitr)   #  install if needed. for showing tables with kable 
 kable( head(readData.out$data) )    # show the first few rows of data 
```

|           | CAX22ROOTFL\_1 | CAX22SHOOTFL\_1 | CAX22ROOTGC\_1 | CAX22SHOOTGC\_1 | CAX22ROOTFL\_2 | CAX22SHOOTFL\_2 | CAX22ROOTGC\_2 | CAX22SHOOTGC\_2 | CAX22ROOTFL\_3 | CAX22SHOOTFL\_3 | CAX22ROOTGC\_3 | CAX22SHOOTGC\_3 | CAX22ROOTFL\_4 | CAX22SHOOTFL\_4 | CAX22ROOTGC\_4 | CAX22SHOOTGC\_4 | CAX23ROOTFL\_4 | CAX23SHOOTFL\_1 | CAX23SHOOTGC\_1 | CAX23ROOTGC\_2 | CAX23ROOTFL\_2 | CAX23SHOOTFL\_2 | CAX23ROOTGC\_1 | CAX23SHOOTGC\_2 | CAX23ROOTFL\_3 | CAX23SHOOTFL\_3 | CAX23ROOTGC\_3 | CAX23SHOOTGC\_3 | CAX23ROOTFL\_1 | CAX23SHOOTFL\_4 | CAX23ROOTGC\_4 | CAX23SHOOTGC\_4 | COL00ROOTFL\_1 | COL00SHOOTFL\_1 | COL00ROOTGC\_1 | COL00SHOOTGC\_1 | COL00ROOTFL\_2 | COL00SHOOTFL\_2 | COL00ROOTGC\_2 | COL00SHOOTGC\_2 | COL00ROOTFL\_3 | COL00SHOOTFL\_3 | COL00ROOTGC\_3 | COL00SHOOTGC\_3 | COL00ROOTFL\_4 | COL00SHOOTFL\_4 | COL00ROOTGC\_4 | COL00SHOOTGC\_4 |
| --------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | --------------- | -------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- | -------------- | --------------- |
| AT3G02260 | 12.289739      | 12.812606       | 12.290257      | 12.708875       | 12.44790       | 12.896278       | 12.481158      | 12.961016       | 12.309401      | 12.804208       | 12.284985      | 12.791365       | 12.313942      | 12.864465       | 12.274680      | 12.936344       | 12.377210      | 12.662967       | 13.059057       | 12.428680      | 12.338665      | 12.834057       | 12.385687      | 12.872359       | 12.255130      | 12.677769       | 12.353087      | 12.883945       | 12.267372      | 12.704204       | 12.918238      | 12.745647       | 12.360296      | 12.838618       | 12.508260      | 13.119420       | 12.288657      | 12.788730       | 12.265305      | 12.734957       | 12.270572      | 12.722722       | 12.342999      | 12.843580       | 12.268932      | 12.586214       | 12.313475      | 12.799549       |
| AT1G67120 | 11.578247      | 11.605370       | 11.792153      | 11.462704       | 11.67525       | 11.254759       | 11.556326      | 11.484208       | 11.729323      | 11.091153       | 11.741650      | 11.425838       | 11.756797      | 11.296305       | 11.664386      | 11.262282       | 11.709118      | 10.888163       | 11.359635       | 11.650400      | 11.595596      | 11.393954       | 11.497300      | 11.488903       | 11.680921      | 11.302084       | 11.613629      | 11.455419       | 11.649615      | 11.172827       | 11.756318      | 10.950964       | 11.530886      | 10.961605       | 11.437625      | 10.546685       | 11.579240      | 11.078471       | 11.494701      | 11.189490       | 11.561420      | 11.017378       | 11.527890      | 10.840132       | 11.620057      | 11.328013       | 11.524369      | 11.055605       |
| AT5G04140 | 8.590108       | 12.333911       | 8.105034       | 12.066482       | 10.13325       | 12.445904       | 7.606691       | 12.613624       | 8.030383       | 12.291843       | 7.952339       | 12.285068       | 7.708635       | 12.257703       | 7.764474       | 12.425695       | 8.017681       | 12.323932       | 12.710464       | 8.879264       | 8.247178       | 12.402466       | 7.813067       | 12.487153       | 8.500349       | 12.244934       | 7.932717       | 12.434338       | 8.089849       | 12.363757       | 6.848955       | 12.252030       | 7.807549       | 12.583546       | 7.198636       | 12.883090       | 8.017914       | 12.377300       | 7.941963       | 12.169942       | 7.323084       | 12.313951       | 7.180027       | 12.378050       | 7.572719       | 12.058312       | 7.750846       | 12.419071       |
| AT5G53460 | 11.748511      | 12.025470       | 11.653364      | 11.938781       | 12.02244       | 12.170971       | 12.016258      | 12.275134       | 11.763959      | 12.021646       | 11.719447      | 11.922042       | 11.789822      | 11.978641       | 11.699979      | 12.136842       | 11.888151      | 11.983322       | 12.304655       | 11.967106      | 11.814141      | 12.019531       | 11.889769      | 12.078651       | 11.691013      | 11.880622       | 11.848031      | 12.017483       | 11.782782      | 11.901540       | 13.058549      | 11.962193       | 11.944182      | 12.004582       | 12.122633      | 12.308480       | 11.823064      | 11.945997       | 11.690854      | 11.917884       | 11.807835      | 12.073534       | 11.871904      | 12.042817       | 11.795714      | 11.832222       | 11.830565      | 12.058742       |
| AT1G70320 | 11.780504      | 12.180105       | 11.830694      | 12.123259       | 11.98443       | 12.252209       | 12.007022      | 12.265655       | 11.831991      | 12.177495       | 11.810753      | 12.147326       | 11.868285      | 12.236467       | 11.797532      | 12.231030       | 11.918702      | 12.123760       | 12.425130       | 11.975768      | 11.887191      | 12.251554       | 11.894194      | 12.271129       | 11.801428      | 12.175849       | 11.883352      | 12.259758       | 11.832488      | 12.168284       | 12.573393      | 12.122780       | 11.869505      | 12.224576       | 12.029739      | 12.385053       | 11.840931      | 12.153916       | 11.762040      | 12.063484       | 11.830963      | 12.135865       | 11.888394      | 12.134324       | 11.826732      | 12.088063       | 11.843371      | 12.199699       |
| AT1G15210 | 10.695619      | 5.019245        | 10.710077      | 6.342336        | 10.89645       | 8.753859        | 10.985095      | 4.924859        | 10.744369      | 7.372438        | 10.686510      | 6.728601        | 10.766530      | 6.025629        | 10.744777      | 5.951211        | 10.783295      | 5.563467        | 5.797079        | 10.986323      | 10.751055      | 3.853542        | 10.894082      | 5.136054        | 10.680861      | 6.743653        | 10.881338      | 5.924098        | 10.699767      | 6.545327        | 11.922253      | 5.602011        | 10.790426      | 6.347902        | 11.112542      | 4.293451        | 10.762160      | 3.899379        | 10.758604      | 6.282959        | 10.775713      | 5.394368        | 10.929811      | 5.503691        | 10.694031      | 7.270401        | 10.860193      | 6.224324        |

```
 readSampleInfo.out <- readSampleInfo(sampleInfoFile) 
 kable( readSampleInfo.out )  
```

|                 | Genotype\_tissue\_treatment | Tissue | Location | Cond | Genotype | Treatment\_genotype | Treatment\_tissue | Genotype\_tissue |
| --------------- | --------------------------- | ------ | -------- | ---- | -------- | ------------------- | ----------------- | ---------------- |
| CAX22ROOTFL\_1  | CAX22ROOTFL                 | ROOT   | A3       | FL   | CAX22    | FLCAX22             | FLROOT            | CAX22ROOT        |
| CAX22SHOOTFL\_1 | CAX22SHOOTFL                | SHOOT  | A3       | FL   | CAX22    | FLCAX22             | FLSHOOT           | CAX22SHOOT       |
| CAX22ROOTGC\_1  | CAX22ROOTGC                 | ROOT   | A3       | GC   | CAX22    | GCCAX22             | GCROOT            | CAX22ROOT        |
| CAX22SHOOTGC\_1 | CAX22SHOOTGC                | SHOOT  | A3       | GC   | CAX22    | GCCAX22             | GCSHOOT           | CAX22SHOOT       |
| CAX22ROOTFL\_2  | CAX22ROOTFL                 | ROOT   | A7       | FL   | CAX22    | FLCAX22             | FLROOT            | CAX22ROOT        |
| CAX22SHOOTFL\_2 | CAX22SHOOTFL                | SHOOT  | A7       | FL   | CAX22    | FLCAX22             | FLSHOOT           | CAX22SHOOT       |
| CAX22ROOTGC\_2  | CAX22ROOTGC                 | ROOT   | A7       | GC   | CAX22    | GCCAX22             | GCROOT            | CAX22ROOT        |
| CAX22SHOOTGC\_2 | CAX22SHOOTGC                | SHOOT  | A7       | GC   | CAX22    | GCCAX22             | GCSHOOT           | CAX22SHOOT       |
| CAX22ROOTFL\_3  | CAX22ROOTFL                 | ROOT   | B4       | FL   | CAX22    | FLCAX22             | FLROOT            | CAX22ROOT        |
| CAX22SHOOTFL\_3 | CAX22SHOOTFL                | SHOOT  | B4       | FL   | CAX22    | FLCAX22             | FLSHOOT           | CAX22SHOOT       |
| CAX22ROOTGC\_3  | CAX22ROOTGC                 | ROOT   | B4       | GC   | CAX22    | GCCAX22             | GCROOT            | CAX22ROOT        |
| CAX22SHOOTGC\_3 | CAX22SHOOTGC                | SHOOT  | B4       | GC   | CAX22    | GCCAX22             | GCSHOOT           | CAX22SHOOT       |
| CAX22ROOTFL\_4  | CAX22ROOTFL                 | ROOT   | B8       | FL   | CAX22    | FLCAX22             | FLROOT            | CAX22ROOT        |
| CAX22SHOOTFL\_4 | CAX22SHOOTFL                | SHOOT  | B8       | FL   | CAX22    | FLCAX22             | FLSHOOT           | CAX22SHOOT       |
| CAX22ROOTGC\_4  | CAX22ROOTGC                 | ROOT   | B8       | GC   | CAX22    | GCCAX22             | GCROOT            | CAX22ROOT        |
| CAX22SHOOTGC\_4 | CAX22SHOOTGC                | SHOOT  | B8       | GC   | CAX22    | GCCAX22             | GCSHOOT           | CAX22SHOOT       |
| CAX23ROOTFL\_4  | CAX23ROOTFL                 | ROOT   | B7       | FL   | CAX23    | FLCAX23             | FLROOT            | CAX23ROOT        |
| CAX23SHOOTFL\_1 | CAX23SHOOTFL                | SHOOT  | A4       | FL   | CAX23    | FLCAX23             | FLSHOOT           | CAX23SHOOT       |
| CAX23SHOOTGC\_1 | CAX23SHOOTGC                | SHOOT  | A4       | GC   | CAX23    | GCCAX23             | GCSHOOT           | CAX23SHOOT       |
| CAX23ROOTGC\_2  | CAX23ROOTGC                 | ROOT   | A8       | GC   | CAX23    | GCCAX23             | GCROOT            | CAX23ROOT        |
| CAX23ROOTFL\_2  | CAX23ROOTFL                 | ROOT   | A8       | FL   | CAX23    | FLCAX23             | FLROOT            | CAX23ROOT        |
| CAX23SHOOTFL\_2 | CAX23SHOOTFL                | SHOOT  | A8       | FL   | CAX23    | FLCAX23             | FLSHOOT           | CAX23SHOOT       |
| CAX23ROOTGC\_1  | CAX23ROOTGC                 | ROOT   | A4       | GC   | CAX23    | GCCAX23             | GCROOT            | CAX23ROOT        |
| CAX23SHOOTGC\_2 | CAX23SHOOTGC                | SHOOT  | A8       | GC   | CAX23    | GCCAX23             | GCSHOOT           | CAX23SHOOT       |
| CAX23ROOTFL\_3  | CAX23ROOTFL                 | ROOT   | B3       | FL   | CAX23    | FLCAX23             | FLROOT            | CAX23ROOT        |
| CAX23SHOOTFL\_3 | CAX23SHOOTFL                | SHOOT  | B3       | FL   | CAX23    | FLCAX23             | FLSHOOT           | CAX23SHOOT       |
| CAX23ROOTGC\_3  | CAX23ROOTGC                 | ROOT   | B3       | GC   | CAX23    | GCCAX23             | GCROOT            | CAX23ROOT        |
| CAX23SHOOTGC\_3 | CAX23SHOOTGC                | SHOOT  | B3       | GC   | CAX23    | GCCAX23             | GCSHOOT           | CAX23SHOOT       |
| CAX23ROOTFL\_1  | CAX23ROOTFL                 | ROOT   | A4       | FL   | CAX23    | FLCAX23             | FLROOT            | CAX23ROOT        |
| CAX23SHOOTFL\_4 | CAX23SHOOTFL                | SHOOT  | B7       | FL   | CAX23    | FLCAX23             | FLSHOOT           | CAX23SHOOT       |
| CAX23ROOTGC\_4  | CAX23ROOTGC                 | ROOT   | B7       | GC   | CAX23    | GCCAX23             | GCROOT            | CAX23ROOT        |
| CAX23SHOOTGC\_4 | CAX23SHOOTGC                | SHOOT  | B7       | GC   | CAX23    | GCCAX23             | GCSHOOT           | CAX23SHOOT       |
| COL00ROOTFL\_1  | COL00ROOTFL                 | ROOT   | A1       | FL   | COL00    | FLCOL00             | FLROOT            | COL00ROOT        |
| COL00SHOOTFL\_1 | COL00SHOOTFL                | SHOOT  | A1       | FL   | COL00    | FLCOL00             | FLSHOOT           | COL00SHOOT       |
| COL00ROOTGC\_1  | COL00ROOTGC                 | ROOT   | A1       | GC   | COL00    | GCCOL00             | GCROOT            | COL00ROOT        |
| COL00SHOOTGC\_1 | COL00SHOOTGC                | SHOOT  | A1       | GC   | COL00    | GCCOL00             | GCSHOOT           | COL00SHOOT       |
| COL00ROOTFL\_2  | COL00ROOTFL                 | ROOT   | B6       | FL   | COL00    | FLCOL00             | FLROOT            | COL00ROOT        |
| COL00SHOOTFL\_2 | COL00SHOOTFL                | SHOOT  | B6       | FL   | COL00    | FLCOL00             | FLSHOOT           | COL00SHOOT       |
| COL00ROOTGC\_2  | COL00ROOTGC                 | ROOT   | B6       | GC   | COL00    | GCCOL00             | GCROOT            | COL00ROOT        |
| COL00SHOOTGC\_2 | COL00SHOOTGC                | SHOOT  | B6       | GC   | COL00    | GCCOL00             | GCSHOOT           | COL00SHOOT       |
| COL00ROOTFL\_3  | COL00ROOTFL                 | ROOT   | B9       | FL   | COL00    | FLCOL00             | FLROOT            | COL00ROOT        |
| COL00SHOOTFL\_3 | COL00SHOOTFL                | SHOOT  | B9       | FL   | COL00    | FLCOL00             | FLSHOOT           | COL00SHOOT       |
| COL00ROOTGC\_3  | COL00ROOTGC                 | ROOT   | B9       | GC   | COL00    | GCCOL00             | GCROOT            | COL00ROOT        |
| COL00SHOOTGC\_3 | COL00SHOOTGC                | SHOOT  | B9       | GC   | COL00    | GCCOL00             | GCSHOOT           | COL00SHOOT       |
| COL00ROOTFL\_4  | COL00ROOTFL                 | ROOT   | C3       | FL   | COL00    | FLCOL00             | FLROOT            | COL00ROOT        |
| COL00SHOOTFL\_4 | COL00SHOOTFL                | SHOOT  | C3       | FL   | COL00    | FLCOL00             | FLSHOOT           | COL00SHOOT       |
| COL00ROOTGC\_4  | COL00ROOTGC                 | ROOT   | C3       | GC   | COL00    | GCCOL00             | GCROOT            | COL00ROOT        |
| COL00SHOOTGC\_4 | COL00SHOOTGC                | SHOOT  | C3       | GC   | COL00    | GCCOL00             | GCSHOOT           | COL00SHOOT       |

```
 input_selectOrg ="NEW" 
 input_selectGO <- 'GOBP'   #Gene set category 
 input_noIDConversion = TRUE  
 allGeneInfo.out <- geneInfo(geneInfoFile) 
 converted.out = NULL 
 convertedData.out <- convertedData()    
 nGenesFilter()  
```

```
## [1] "25848 genes in 48 samples. 25848  genes passed filter.\n Original gene IDs used."
```

```
 convertedCounts.out <- convertedCounts()  # converted counts, just for compatibility 
```

### 2. Pre-process

```
# Read counts per library 
 parDefault = par() 
 par(mar=c(12,4,2,2)) 
 # barplot of total read counts
 x <- readData.out$rawCounts
 groups = as.factor( detectGroups(colnames(x ) ) )
 if(nlevels(groups)<=1 | nlevels(groups) >20 )  
  col1 = 'green'  else
  col1 = rainbow(nlevels(groups))[ groups ]             
         
 barplot( colSums(x)/1e6, 
        col=col1,las=3, main="Total read counts (millions)")  
```

```
 readCountsBias()  # detecting bias in sequencing depth 
```

```
## [1] "Warning! Sequencing depth bias detected. Total read counts are significantly different among sample groups (p= 7.17e-08 ) based on ANOVA.  Total read counts seem to be correlated with factor Genotype_tissue_treatment (p= 7.17e-08 ).    Total read counts seem to be correlated with factor Tissue (p= 1.16e-12 ).    Total read counts seem to be correlated with factor Treatment_tissue (p= 3.69e-11 ).    Total read counts seem to be correlated with factor Genotype_tissue (p= 3.11e-10 ).  "
```

```
 # Box plot 
 x = readData.out$data 
 boxplot(x, las = 2, col=col1,
    ylab='Transformed expression levels',
    main='Distribution of transformed data') 
```

```
 #Density plot 
 par(parDefault) 
```

```
## Warning in par(parDefault): graphical parameter "cin" cannot be set
```

```
## Warning in par(parDefault): graphical parameter "cra" cannot be set
```

```
## Warning in par(parDefault): graphical parameter "csi" cannot be set
```

```
## Warning in par(parDefault): graphical parameter "cxy" cannot be set
```

```
## Warning in par(parDefault): graphical parameter "din" cannot be set
```

```
## Warning in par(parDefault): graphical parameter "page" cannot be set
```

```
 densityPlot()       
```

```
 # Scatter plot of the first two samples 
 plot(x[,1:2],xlab=colnames(x)[1],ylab=colnames(x)[2], 
    main='Scatter plot of first two samples') 
```

### 3. Heatmap

```
 # hierarchical clustering tree
 x <- readData.out$data
 maxGene <- apply(x,1,max)
 # remove bottom 25% lowly expressed genes, which inflate the PPC
 x <- x[which(maxGene > quantile(maxGene)[1] ) ,] 
 plot(as.dendrogram(hclust2( dist2(t(x)))), ylab="1 - Pearson C.C.", type = "rectangle") 
```

```
 #Correlation matrix
 input_labelPCC <- TRUE #Show correlation coefficient? 
 correlationMatrix() 
```

```
## Warning in melt(round(cor(x), 2), na.rm = TRUE): The melt generic in data.table
## has been passed a matrix and will attempt to redirect to the relevant reshape2
## method; please note that reshape2 is deprecated, and this redirection is now
## deprecated as well. To continue using melt methods from reshape2 while both
## libraries are attached, e.g. melt.list, you can prepend the namespace like
## reshape2::melt(round(cor(x), 2)). In the next version, this warning will become
## an error.
```

```
## Warning: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> =
## "none")` instead.
```

```
 # Parameters for heatmap
 input_nGenes <- 1000   #Top genes for heatmap
 input_geneCentering <- TRUE    #centering genes ?
 input_sampleCentering <- TRUE  #Center by sample?
 input_geneNormalize <- TRUE    #Normalize by gene?
 input_sampleNormalize <- TRUE  #Normalize by sample?
 input_noSampleClustering <- FALSE  #Use original sample order
 input_heatmapCutoff <- 4   #Remove outliers beyond number of SDs 
 input_distFunctions <- 1   #which distant funciton to use
 input_hclustFunctions <- 1 #Linkage type
 input_heatColors1 <- 2 #Colors
 input_selectFactorsHeatmap <- 'Genotype_tissue_treatment'  #Sample coloring factors 
 png('heatmap.png', width = 10, height = 15, units = 'in', res = 300) 
 staticHeatmap() 
 dev.off()  
```

```
## quartz_off_screen 
##                 2
```

\[heatmap] (heatmap.png)

```
 heatmapPlotly() # interactive heatmap using Plotly 
```

```
## Warning in melt(.): The melt generic in data.table has been passed a matrix
## and will attempt to redirect to the relevant reshape2 method; please note that
## reshape2 is deprecated, and this redirection is now deprecated as well. To
## continue using melt methods from reshape2 while both libraries are attached,
## e.g. melt.list, you can prepend the namespace like reshape2::melt(.). In the
## next version, this warning will become an error.
```

### 4. K-means clustering

```
 input_nGenesKNN <- 2000    #Number of genes fro k-Means
 input_nClusters <- 4   #Number of clusters 
 maxGeneClustering = 12000
 input_kmeansNormalization <- 'geneMean'    #Normalization
 input_KmeansReRun <- 0 #Random seed 

 distributionSD()  #Distribution of standard deviations 
```

```
 KmeansNclusters()  #Number of clusters 
```

```
 Kmeans.out = Kmeans()   #Running K-means 
 KmeansHeatmap()   #Heatmap for k-Means 
```

```
 #Read gene sets for enrichment analysis 
 sqlite  <- dbDriver('SQLite')
 input_selectGO3 <- 'GOBP'  #Gene set category
 input_minSetSize <- 5  #Min gene set size
 input_maxSetSize <- 2000   #Max gene set size 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO3,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  )  
 # Alternatively, users can use their own GMT files by
 #GeneSets.out <- readGMTRobust('somefile.GMT')  
 results <- KmeansGO()  #Enrichment analysis for k-Means clusters   
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Cluster | adj.Pval | Genes | Pathways                                          |
| ------- | -------- | ----- | ------------------------------------------------- |
| A       | 3.74e-19 | 19    | Microtubule-based movement                        |
|         | 2.49e-18 | 20    | Movement of cell or subcellular component         |
|         | 4.18e-18 | 28    | Microtubule-based process                         |
|         | 1.03e-13 | 45    | Anatomical structure morphogenesis                |
|         | 1.51e-09 | 32    | Cell cycle                                        |
|         | 3.07e-09 | 14    | Cytokinesis                                       |
|         | 3.24e-09 | 50    | Phosphorylation                                   |
|         | 5.52e-09 | 26    | Cell cycle process                                |
|         | 1.52e-08 | 27    | External encapsulating structure organization     |
|         | 2.83e-08 | 41    | Protein phosphorylation                           |
| B       | 2.37e-09 | 11    | Phenylpropanoid metabolic process                 |
|         | 2.37e-09 | 12    | Cellular response to toxic substance              |
|         | 2.37e-09 | 12    | Cellular detoxification                           |
|         | 1.64e-08 | 11    | Cellular oxidant detoxification                   |
|         | 1.89e-08 | 9     | Hydrogen peroxide catabolic process               |
|         | 2.20e-08 | 12    | Detoxification                                    |
|         | 9.06e-08 | 12    | Response to toxic substance                       |
|         | 9.06e-08 | 9     | Hydrogen peroxide metabolic process               |
|         | 9.38e-08 | 13    | Secondary metabolic process                       |
|         | 3.99e-06 | 9     | Reactive oxygen species metabolic process         |
| C       | 4.53e-56 | 49    | Photosynthesis                                    |
|         | 7.88e-32 | 28    | Photosynthesis, light reaction                    |
|         | 1.48e-29 | 39    | Generation of precursor metabolites and energy    |
|         | 1.26e-28 | 16    | Photosynthesis, light harvesting in photosystem I |
|         | 4.80e-25 | 40    | Response to light stimulus                        |
|         | 1.37e-24 | 40    | Response to radiation                             |
|         | 2.02e-24 | 16    | Photosynthesis, light harvesting                  |
|         | 1.65e-22 | 53    | Response to external stimulus                     |
|         | 1.98e-22 | 17    | Protein-chromophore linkage                       |
|         | 4.45e-20 | 44    | Response to external biotic stimulus              |
| D       | 1.83e-56 | 182   | Cellular localization                             |
|         | 5.94e-48 | 190   | Macromolecule localization                        |
|         | 5.94e-48 | 145   | Intracellular transport                           |
|         | 7.05e-46 | 147   | Establishment of localization in cell             |
|         | 2.13e-45 | 93    | Response to cadmium ion                           |
|         | 6.65e-43 | 159   | Protein localization                              |
|         | 1.57e-41 | 103   | Response to metal ion                             |
|         | 1.56e-40 | 130   | Cellular macromolecule localization               |
|         | 1.56e-39 | 222   | Organelle organization                            |
|         | 1.31e-37 | 122   | Cellular protein localization                     |

```
 input_seedTSNE <- 0    #Random seed for t-SNE
 input_colorGenes <- TRUE   #Color genes in t-SNE plot? 
 tSNEgenePlot()  #Plot genes using t-SNE 
```

### 5. PCA and beyond

```
 input_selectFactors <- 'Genotype_tissue_treatment' #Factor coded by color
 input_selectFactors2 <- 'Treatment_tissue' #Factor coded by shape
 input_tsneSeed2 <- 0   #Random seed for t-SNE 
 #PCA, MDS and t-SNE plots
 PCAplot()  
```

```
 MDSplot() 
```

```
 tSNEplot()  
```

```
 #Read gene sets for pathway analysis using PGSEA on principal components
 input_selectGO6 <- 'Curated.AraCyc'    #Gene set category 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO6,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  )  
 PCApathway() # Run PGSEA analysis 
```

```
 cat( PCA2factor() )   #The correlation between PCs with factors 
```

```
## 
##  Correlation between Principal Components (PCs) with factors
## PC1 is correlated with Tissue (p=1.36e-50).
## PC2 is correlated with Treatment_genotype (p=7.84e-07).
## PC4 is correlated with Treatment_genotype (p=4.91e-05).
```

### 6. DEG1

```
 input_CountsDEGMethod <- 3 #DESeq2= 3,limma-voom=2,limma-trend=1 
 input_limmaPval <- 0.1 #FDR cutoff
 input_limmaFC <- 2 #Fold-change cutoff
 input_selectModelComprions <- c('CAX22ROOTFL-CAX22ROOTGC','CAX22ROOTGC-COL00ROOTGC','CAX22SHOOTFL-CAX22SHOOTGC','CAX22SHOOTGC-COL00SHOOTGC','CAX23ROOTFL-CAX23ROOTGC','CAX23ROOTGC-COL00ROOTGC','CAX23SHOOTFL-CAX23SHOOTGC','CAX23SHOOTGC-COL00SHOOTGC','COL00ROOTFL-COL00ROOTGC','COL00SHOOTFL-COL00SHOOTGC') #Selected comparisons
 input_selectFactorsModel <- NULL   #Selected comparisons
 input_selectInteractions <- NULL   #Selected comparisons
 input_selectBlockFactorsModel <- NULL  #Selected comparisons 
 factorReferenceLevels.out <- NULL 

 limma.out <- limma()
```

```
## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors
```

```
 DEG.data.out <- DEG.data()
 limma.out$comparisons 
```

```
##  [1] "CAX22ROOTFL-CAX22ROOTGC"   "CAX22ROOTGC-COL00ROOTGC"  
##  [3] "CAX22SHOOTFL-CAX22SHOOTGC" "CAX22SHOOTGC-COL00SHOOTGC"
##  [5] "CAX23ROOTFL-CAX23ROOTGC"   "CAX23ROOTGC-COL00ROOTGC"  
##  [7] "CAX23SHOOTFL-CAX23SHOOTGC" "CAX23SHOOTGC-COL00SHOOTGC"
##  [9] "COL00ROOTFL-COL00ROOTGC"   "COL00SHOOTFL-COL00SHOOTGC"
```

Venn diagram for root comparison

```
 input_selectComparisonsVenn <- c('CAX22ROOTFL-CAX22ROOTGC','CAX22ROOTGC-COL00ROOTGC','CAX23ROOTFL-CAX23ROOTGC','CAX23ROOTGC-COL00ROOTGC','COL00ROOTFL-COL00ROOTGC')    #Selected comparisons for Venn diagram
 input_UpDownRegulated <- FALSE #Split up and down regulated genes 
 vennPlot() # Venn diagram 
```

Venn diagram for shoot comparison

```
 input_selectComparisonsVenn <- c('CAX22SHOOTFL-CAX22SHOOTGC','CAX22SHOOTGC-COL00SHOOTGC','CAX23SHOOTFL-CAX23SHOOTGC','CAX23SHOOTGC-COL00SHOOTGC','COL00SHOOTFL-COL00SHOOTGC')  #Selected comparisons for Venn diagram
 input_UpDownRegulated <- FALSE #Split up and down regulated genes 
 vennPlot() # Venn diagram 
```

```
  sigGeneStats() # number of DEGs as figure 
```

```
## Warning in melt(stats): The melt generic in data.table has been passed a matrix
## and will attempt to redirect to the relevant reshape2 method; please note that
## reshape2 is deprecated, and this redirection is now deprecated as well. To
## continue using melt methods from reshape2 while both libraries are attached,
## e.g. melt.list, you can prepend the namespace like reshape2::melt(stats). In the
## next version, this warning will become an error.
```

```
  sigGeneStatsTable() # number of DEGs as table 
```

```
##                                         Comparisons  Up Down
## CAX22ROOTFL-CAX22ROOTGC     CAX22ROOTFL-CAX22ROOTGC 180    0
## CAX22ROOTGC-COL00ROOTGC     CAX22ROOTGC-COL00ROOTGC  12  113
## CAX22SHOOTFL-CAX22SHOOTGC CAX22SHOOTFL-CAX22SHOOTGC   7    4
## CAX22SHOOTGC-COL00SHOOTGC CAX22SHOOTGC-COL00SHOOTGC  53  121
## CAX23ROOTFL-CAX23ROOTGC     CAX23ROOTFL-CAX23ROOTGC 352  459
## CAX23ROOTGC-COL00ROOTGC     CAX23ROOTGC-COL00ROOTGC  16   13
## CAX23SHOOTFL-CAX23SHOOTGC CAX23SHOOTFL-CAX23SHOOTGC 325   45
## CAX23SHOOTGC-COL00SHOOTGC CAX23SHOOTGC-COL00SHOOTGC  18  143
## COL00ROOTFL-COL00ROOTGC     COL00ROOTFL-COL00ROOTGC 204   90
## COL00SHOOTFL-COL00SHOOTGC COL00SHOOTFL-COL00SHOOTGC 299   33
```

### 7. DEG2.1

```
 input_selectContrast <- 'COL00SHOOTFL-COL00SHOOTGC'    #Selected comparisons 
 selectedHeatmap.data.out <- selectedHeatmap.data()
 selectedHeatmap()   # heatmap for DEGs in selected comparison
```

```
 # Save gene lists and data into files
 write.csv( selectedHeatmap.data()$genes, 'heatmap.data.csv') 
 write.csv(DEG.data(),'DEG.data.csv' )
 write(AllGeneListsGMT() ,'AllGeneListsGMT.gmt')
```

```
 input_selectGO2 <- 'GOBP'  #Gene set category 
 geneListData.out <- geneListData()  
 volcanoPlot()  
```

```
  scatterPlot()  
```

```
  MAplot()  
```

```
  geneListGOTable.out <- geneListGOTable()  
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO2,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  ) 
 input_removeRedudantSets <- TRUE   #Remove highly redundant gene sets? 
 results <- geneListGO()  #Enrichment analysis
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction    | adj.Pval | nGenes | Pathways                               |
| ------------ | -------- | ------ | -------------------------------------- |
| Up regulated | 2.3e-03  | 4      | Chitin degradation II                  |
|              | 3.1e-03  | 6      | Glutathione-mediated detoxification II |

### 7. DEG2.2

```
 input_selectContrast <- 'COL00ROOTFL-COL00ROOTGC'  #Selected comparisons 
 selectedHeatmap.data.out <- selectedHeatmap.data()
 selectedHeatmap()   # heatmap for DEGs in selected comparison
```

```
 # Save gene lists and data into files
 write.csv( selectedHeatmap.data()$genes, 'heatmap.data.csv') 
 write.csv(DEG.data(),'DEG.data.csv' )
 write(AllGeneListsGMT() ,'AllGeneListsGMT.gmt')
```

```
 input_selectGO2 <- 'GOBP'  #Gene set category 
 geneListData.out <- geneListData()  
 volcanoPlot()  
```

```
  scatterPlot()  
```

```
  MAplot()  
```

```
  geneListGOTable.out <- geneListGOTable()  
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO2,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  ) 
 input_removeRedudantSets <- TRUE   #Remove highly redundant gene sets? 
 results <- geneListGO()  #Enrichment analysis
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction    | adj.Pval | nGenes | Pathways                                     |
| ------------ | -------- | ------ | -------------------------------------------- |
| Up regulated | 4.3e-09  | 18     | Response to toxic substance                  |
|              | 2.8e-06  | 14     | Detoxification                               |
|              | 5.4e-06  | 17     | Sulfur compound metabolic process            |
|              | 6.0e-06  | 32     | Defense response                             |
|              | 6.0e-06  | 13     | Cellular response to decreased oxygen levels |
|              | 6.0e-06  | 13     | Cellular response to oxygen levels           |
|              | 6.0e-06  | 13     | Cellular response to hypoxia                 |
|              | 1.1e-05  | 13     | Response to hypoxia                          |
|              | 1.1e-05  | 25     | Immune system process                        |
|              | 1.1e-05  | 25     | Immune response                              |

### 7. DEG2.3

```
 input_selectContrast <- 'CAX22ROOTFL-CAX22ROOTGC'  #Selected comparisons 
 selectedHeatmap.data.out <- selectedHeatmap.data()
 selectedHeatmap()   # heatmap for DEGs in selected comparison
```

```
 # Save gene lists and data into files
 write.csv( selectedHeatmap.data()$genes, 'heatmap.data.csv') 
 write.csv(DEG.data(),'DEG.data.csv' )
 write(AllGeneListsGMT() ,'AllGeneListsGMT.gmt')
```

```
 input_selectGO2 <- 'GOBP'  #Gene set category 
 geneListData.out <- geneListData()  
 volcanoPlot()  
```

```
  scatterPlot()  
```

```
  MAplot()  
```

```
  geneListGOTable.out <- geneListGOTable()  
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO2,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  ) 
 input_removeRedudantSets <- TRUE   #Remove highly redundant gene sets? 
 results <- geneListGO()  #Enrichment analysis
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction    | adj.Pval | nGenes | Pathways                                          |
| ------------ | -------- | ------ | ------------------------------------------------- |
| Up regulated | 2.6e-04  | 23     | Lipid metabolic process                           |
|              | 2.6e-04  | 5      | Photosynthesis, light harvesting in photosystem I |
|              | 2.6e-04  | 12     | Photosynthesis                                    |
|              | 2.6e-04  | 11     | Lipid catabolic process                           |
|              | 3.0e-04  | 18     | Response to light stimulus                        |
|              | 4.0e-04  | 18     | Response to radiation                             |
|              | 1.2e-03  | 5      | Photosynthesis, light harvesting                  |
|              | 1.2e-03  | 8      | Photosynthesis, light reaction                    |

### 7. DEG2.4

```
 input_selectContrast <- 'CAX22SHOOTFL-CAX22SHOOTGC'    #Selected comparisons 
 selectedHeatmap.data.out <- selectedHeatmap.data()
 selectedHeatmap()   # heatmap for DEGs in selected comparison
```

```
 # Save gene lists and data into files
 write.csv( selectedHeatmap.data()$genes, 'heatmap.data.csv') 
 write.csv(DEG.data(),'DEG.data.csv' )
 write(AllGeneListsGMT() ,'AllGeneListsGMT.gmt')
```

```
 input_selectGO2 <- 'GOBP'  #Gene set category 
 geneListData.out <- geneListData()  
 volcanoPlot()  
```

```
  scatterPlot()  
```

```
  MAplot()  
```

### 7. DEG2.5

```
 input_selectContrast <- 'CAX23ROOTFL-CAX23ROOTGC'  #Selected comparisons 
 selectedHeatmap.data.out <- selectedHeatmap.data()
 selectedHeatmap()   # heatmap for DEGs in selected comparison
```

```
 # Save gene lists and data into files
 write.csv( selectedHeatmap.data()$genes, 'heatmap.data.csv') 
 write.csv(DEG.data(),'DEG.data.csv' )
 write(AllGeneListsGMT() ,'AllGeneListsGMT.gmt')
```

```
 input_selectGO2 <- 'GOBP'  #Gene set category 
 geneListData.out <- geneListData()  
 volcanoPlot()  
```

```
  scatterPlot()  
```

```
  MAplot()  
```

### 7. DEG2.6

```
 input_selectContrast <- 'CAX23SHOOTFL-CAX23SHOOTGC'    #Selected comparisons 
 selectedHeatmap.data.out <- selectedHeatmap.data()
 selectedHeatmap()   # heatmap for DEGs in selected comparison
```

```
 # Save gene lists and data into files
 write.csv( selectedHeatmap.data()$genes, 'heatmap.data.csv') 
 write.csv(DEG.data(),'DEG.data.csv' )
 write(AllGeneListsGMT() ,'AllGeneListsGMT.gmt')
```

```
 input_selectGO2 <- 'GOBP'  #Gene set category 
 geneListData.out <- geneListData()  
 volcanoPlot()  
```

```
  scatterPlot()  
```

```
  MAplot()  
```

### 8.1 Pathway analysis

```
 input_selectContrast1 <- 'COL00SHOOTFL-COL00SHOOTGC'   #select Comparison 
 #input_selectContrast1 = limma.out$comparisons[3] # manually set
 input_selectGO <- 'GOBP'   #Gene set category 
 #input_selectGO='custom' # if custom gmt file
 input_minSetSize <- 5  #Min size for gene set
 input_maxSetSize <- 2000   #Max size for gene set 
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  ) 
 input_pathwayPvalCutoff <- 0.2 #FDR cutoff
 input_nPathwayShow <- 30   #Top pathways to show
 input_absoluteFold <- FALSE    #Use absolute values of fold-change?
 input_GenePvalCutoff <- 1  #FDR to remove genes 

 input_pathwayMethod = 1  # 1  GAGE
 gagePathwayData.out <- gagePathwayData()  # pathway analysis using GAGE  
   
 results <- gagePathwayData.out  #Enrichment analysis for k-Means clusters  
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction | GAGE analysis: COL00SHOOTFL vs COL00SHOOTGC | statistic | Genes | adj.Pval |
| --------- | ------------------------------------------- | --------- | ----- | -------- |
| Down      | Photosynthesis                              | -3.8464   | 265   | 1.7e-01  |
| Up        | Response to toxic substance                 | 4.4428    | 259   | 9.1e-03  |
|           | Detoxification                              | 4.3909    | 221   | 9.1e-03  |
|           | Response to bacterium                       | 3.6581    | 499   | 1.1e-01  |
|           | Toxin metabolic process                     | 3.6109    | 54    | 1.3e-01  |
|           | Toxin catabolic process                     | 3.4758    | 42    | 1.5e-01  |
|           | Defense response to bacterium               | 3.4643    | 429   | 1.3e-01  |
|           | Systemic acquired resistance                | 3.3581    | 80    | 1.5e-01  |
|           | Response to extracellular stimulus          | 3.3567    | 248   | 1.5e-01  |
|           | Response to nutrient levels                 | 3.2934    | 212   | 1.5e-01  |
|           | Hydrogen peroxide catabolic process         | 3.2067    | 83    | 1.8e-01  |
|           | Glutathione metabolic process               | 3.1918    | 75    | 1.8e-01  |
|           | Hydrogen peroxide metabolic process         | 3.174     | 104   | 1.8e-01  |
|           | Response to salicylic acid                  | 3.0782    | 161   | 1.8e-01  |
|           | Response to starvation                      | 3.0691    | 182   | 1.8e-01  |
|           | Cellular detoxification                     | 3.0497    | 171   | 1.8e-01  |
|           | Secondary metabolic process                 | 3.0481    | 323   | 1.8e-01  |
|           | Response to fungus                          | 3.0446    | 398   | 1.8e-01  |
|           | Cellular response to toxic substance        | 3.0079    | 172   | 1.9e-01  |

```
 pathwayListData.out = pathwayListData() 
 enrichmentPlot(pathwayListData.out, 25  ) 
```

```
  enrichmentNetwork(pathwayListData.out )  
```

```
  enrichmentNetworkPlotly(pathwayListData.out) 
```

```
 input_pathwayMethod = 3  # 1  fgsea 
 fgseaPathwayData.out <- fgseaPathwayData() #Pathway analysis using fgsea 
```

```
## Warning in fgsea(pathways = gmt, stats = fold, minSize = input_minSetSize, :
## You are trying to run fgseaSimple. It is recommended to use fgseaMultilevel. To
## run fgseaMultilevel, you need to remove the nperm argument in the fgsea function
## call.
```

```
## Warning in fgseaSimple(...): There were 1 pathways for which P-values were not
## calculated properly due to unbalanced gene-level statistic values
```

```
 results <- fgseaPathwayData.out  #Enrichment analysis for k-Means clusters 
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction | GSEA analysis: COL00SHOOTFL vs COL00SHOOTGC          | NES     | Genes | adj.Pval |
| --------- | ---------------------------------------------------- | ------- | ----- | -------- |
| Down      | Leucine metabolic process                            | -1.9004 | 24    | 6.3e-02  |
|           | Leucine biosynthetic process                         | -1.8053 | 15    | 1.4e-01  |
|           | Starch catabolic process                             | -1.7794 | 17    | 1.8e-01  |
| Up        | Toxin metabolic process                              | 1.9401  | 54    | 1.1e-02  |
|           | Glutathione metabolic process                        | 1.937   | 75    | 1.1e-02  |
|           | Toxin catabolic process                              | 1.916   | 42    | 1.1e-02  |
|           | Cellular response to nitrogen compound               | 1.9033  | 63    | 1.1e-02  |
|           | Detoxification                                       | 1.9008  | 221   | 1.1e-02  |
|           | Hydrogen peroxide catabolic process                  | 1.8976  | 83    | 1.1e-02  |
|           | Cellular modified amino acid metabolic process       | 1.8896  | 121   | 1.1e-02  |
|           | Hydrogen peroxide metabolic process                  | 1.8747  | 104   | 1.1e-02  |
|           | Response to toxic substance                          | 1.8712  | 259   | 1.1e-02  |
|           | Response to iron ion starvation                      | 1.8525  | 12    | 1.3e-02  |
|           | Cellular response to organonitrogen compound         | 1.821   | 48    | 1.1e-02  |
|           | Systemic acquired resistance                         | 1.8129  | 80    | 1.1e-02  |
|           | Glucosamine-containing compound metabolic process    | 1.7958  | 24    | 1.9e-02  |
|           | Positive regulation of innate immune response        | 1.7942  | 50    | 1.1e-02  |
|           | Aminoglycan metabolic process                        | 1.793   | 21    | 1.2e-02  |
|           | Cellular detoxification                              | 1.7909  | 171   | 1.1e-02  |
|           | Activation of innate immune response                 | 1.7907  | 40    | 2.2e-02  |
|           | Activation of immune response                        | 1.7907  | 40    | 2.2e-02  |
|           | Positive regulation of response to external stimulus | 1.7833  | 89    | 1.1e-02  |
|           | Cellular response to toxic substance                 | 1.7795  | 172   | 1.1e-02  |
|           | Cellular oxidant detoxification                      | 1.7788  | 158   | 1.1e-02  |
|           | Response to oomycetes                                | 1.7722  | 100   | 1.1e-02  |
|           | Amino sugar metabolic process                        | 1.7659  | 36    | 3.1e-02  |
|           | Response to nutrient levels                          | 1.7654  | 212   | 1.1e-02  |
|           | Cellular response to acid chemical                   | 1.7632  | 56    | 1.7e-02  |
|           | Response to amino acid                               | 1.7622  | 30    | 2.7e-02  |
|           | Response to starvation                               | 1.7612  | 182   | 1.1e-02  |

```
  pathwayListData.out = pathwayListData() 
 enrichmentPlot(pathwayListData.out, 25  ) 
```

```
  enrichmentNetwork(pathwayListData.out )  
```

```
  enrichmentNetworkPlotly(pathwayListData.out) 
```

```
   PGSEAplot() # pathway analysis using PGSEA 
```

```
## 
## Computing P values using ANOVA
```

### 8.2 Pathway analysis

```
 input_selectContrast1 <- 'COL00ROOTFL-COL00ROOTGC' #select Comparison 
 #input_selectContrast1 = limma.out$comparisons[3] # manually set
 input_selectGO <- 'GOBP'   #Gene set category 
 #input_selectGO='custom' # if custom gmt file
 input_minSetSize <- 5  #Min size for gene set
 input_maxSetSize <- 2000   #Max size for gene set 
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  ) 
 input_pathwayPvalCutoff <- 0.2 #FDR cutoff
 input_nPathwayShow <- 30   #Top pathways to show
 input_absoluteFold <- FALSE    #Use absolute values of fold-change?
 input_GenePvalCutoff <- 1  #FDR to remove genes 

 input_pathwayMethod = 1  # 1  GAGE
 gagePathwayData.out <- gagePathwayData()  # pathway analysis using GAGE  
   
 results <- gagePathwayData.out  #Enrichment analysis for k-Means clusters  
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction | GAGE analysis: COL00ROOTFL vs COL00ROOTGC                                 | statistic | Genes | adj.Pval |
| --------- | ------------------------------------------------------------------------- | --------- | ----- | -------- |
| Up        | Biological process involved in interspecies interaction between organisms | 3.8904    | 478   | 2.6e-02  |
|           | Response to external biotic stimulus                                      | 3.8625    | 476   | 2.6e-02  |
|           | Response to other organism                                                | 3.8625    | 476   | 2.6e-02  |
|           | Sulfur compound metabolic process                                         | 3.7023    | 161   | 2.6e-02  |
|           | Cellular response to decreased oxygen levels                              | 3.6693    | 102   | 2.6e-02  |
|           | Cellular response to oxygen levels                                        | 3.6693    | 102   | 2.6e-02  |
|           | Cellular response to hypoxia                                              | 3.6489    | 101   | 2.6e-02  |
|           | Response to decreased oxygen levels                                       | 3.6233    | 109   | 2.6e-02  |
|           | Response to oxygen levels                                                 | 3.6233    | 109   | 2.6e-02  |
|           | Response to biotic stimulus                                               | 3.6052    | 489   | 2.6e-02  |
|           | Response to hypoxia                                                       | 3.603     | 108   | 2.6e-02  |
|           | Response to toxic substance                                               | 3.5744    | 128   | 2.6e-02  |
|           | Response to bacterium                                                     | 3.5665    | 204   | 2.6e-02  |
|           | Toxin metabolic process                                                   | 3.5232    | 31    | 4.0e-02  |
|           | Immune response                                                           | 3.5101    | 369   | 2.7e-02  |
|           | Immune system process                                                     | 3.4885    | 370   | 2.7e-02  |
|           | Defense response to other organism                                        | 3.4659    | 366   | 2.8e-02  |
|           | Cellular response to stress                                               | 3.4192    | 454   | 3.0e-02  |
|           | Secondary metabolic process                                               | 3.2392    | 158   | 5.6e-02  |
|           | Glutathione metabolic process                                             | 3.2379    | 44    | 6.4e-02  |
|           | Toxin catabolic process                                                   | 3.2095    | 29    | 8.0e-02  |
|           | Cellular catabolic process                                                | 3.1727    | 455   | 6.1e-02  |
|           | Response to inorganic substance                                           | 3.15      | 329   | 6.4e-02  |
|           | Detoxification                                                            | 3.1471    | 112   | 6.5e-02  |
|           | Systemic acquired resistance                                              | 3.0886    | 31    | 1.0e-01  |
|           | Oxoacid metabolic process                                                 | 3.0278    | 412   | 8.0e-02  |
|           | Organic acid metabolic process                                            | 3.0087    | 426   | 8.1e-02  |
|           | Response to osmotic stress                                                | 2.887     | 195   | 1.1e-01  |
|           | Response to oxidative stress                                              | 2.8787    | 171   | 1.1e-01  |
|           | Defense response to bacterium                                             | 2.8327    | 170   | 1.3e-01  |

```
 pathwayListData.out = pathwayListData() 
 enrichmentPlot(pathwayListData.out, 25  ) 
```

```
  enrichmentNetwork(pathwayListData.out )  
```

```
  enrichmentNetworkPlotly(pathwayListData.out) 
```

```
 input_pathwayMethod = 3  # 1  fgsea 
 fgseaPathwayData.out <- fgseaPathwayData() #Pathway analysis using fgsea 
```

```
## Warning in fgsea(pathways = gmt, stats = fold, minSize = input_minSetSize, :
## You are trying to run fgseaSimple. It is recommended to use fgseaMultilevel. To
## run fgseaMultilevel, you need to remove the nperm argument in the fgsea function
## call.
```

```
 results <- fgseaPathwayData.out  #Enrichment analysis for k-Means clusters 
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction | GSEA analysis: COL00ROOTFL vs COL00ROOTGC                 | NES     | Genes | adj.Pval |
| --------- | --------------------------------------------------------- | ------- | ----- | -------- |
| Down      | Positive regulation of transcription by RNA polymerase II | -1.9528 | 70    | 2.2e-02  |
|           | Lipid storage                                             | -1.8352 | 10    | 8.5e-02  |
| Up        | Cellular response to hypoxia                              | 2.042   | 101   | 9.4e-03  |
|           | Toxin metabolic process                                   | 2.0388  | 31    | 9.8e-03  |
|           | Cellular response to decreased oxygen levels              | 2.0343  | 102   | 9.4e-03  |
|           | Cellular response to oxygen levels                        | 2.0343  | 102   | 9.4e-03  |
|           | Response to hypoxia                                       | 2.0073  | 108   | 9.4e-03  |
|           | Response to decreased oxygen levels                       | 1.9998  | 109   | 9.4e-03  |
|           | Response to oxygen levels                                 | 1.9998  | 109   | 9.4e-03  |
|           | Toxin catabolic process                                   | 1.9901  | 29    | 9.8e-03  |
|           | Glutathione metabolic process                             | 1.9732  | 44    | 9.8e-03  |
|           | Sulfur compound metabolic process                         | 1.9216  | 161   | 9.4e-03  |
|           | Response to toxic substance                               | 1.9037  | 128   | 9.4e-03  |
|           | Systemic acquired resistance                              | 1.9028  | 31    | 9.8e-03  |
|           | Response to bacterium                                     | 1.8943  | 204   | 9.4e-03  |
|           | Positive regulation of response to external stimulus      | 1.8621  | 38    | 1.7e-02  |
|           | Secondary metabolic process                               | 1.8551  | 158   | 9.4e-03  |
|           | Positive regulation of innate immune response             | 1.843   | 22    | 1.8e-02  |
|           | S-glycoside metabolic process                             | 1.8425  | 43    | 2.2e-02  |
|           | Glycosinolate metabolic process                           | 1.8425  | 43    | 2.2e-02  |
|           | Glucosinolate metabolic process                           | 1.8425  | 43    | 2.2e-02  |
|           | Cellular modified amino acid metabolic process            | 1.8404  | 59    | 9.8e-03  |
|           | Sulfur compound catabolic process                         | 1.8375  | 21    | 3.0e-02  |
|           | Detoxification                                            | 1.83    | 112   | 1.5e-02  |
|           | Glycosyl compound catabolic process                       | 1.8035  | 19    | 5.5e-02  |
|           | Branched-chain amino acid metabolic process               | 1.7962  | 13    | 8.5e-02  |
|           | Response to salicylic acid                                | 1.7888  | 65    | 2.2e-02  |
|           | Response to salt stress                                   | 1.7789  | 166   | 9.4e-03  |
|           | Defense response to bacterium                             | 1.7757  | 170   | 9.4e-03  |
|           | Positive regulation of immune system process              | 1.765   | 36    | 6.9e-02  |

```
  pathwayListData.out = pathwayListData() 
 enrichmentPlot(pathwayListData.out, 25  ) 
```

```
  enrichmentNetwork(pathwayListData.out )  
```

```
  enrichmentNetworkPlotly(pathwayListData.out) 
```

```
   PGSEAplot() # pathway analysis using PGSEA 
```

```
## 
## Computing P values using ANOVA
```

### 8.3 Pathway analysis

```
 input_selectContrast1 <- 'CAX22SHOOTFL-CAX22SHOOTGC'   #select Comparison 
 #input_selectContrast1 = limma.out$comparisons[3] # manually set
 input_selectGO <- 'GOBP'   #Gene set category 
 #input_selectGO='custom' # if custom gmt file
 input_minSetSize <- 5  #Min size for gene set
 input_maxSetSize <- 2000   #Max size for gene set 
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  ) 
 input_pathwayPvalCutoff <- 0.2 #FDR cutoff
 input_nPathwayShow <- 30   #Top pathways to show
 input_absoluteFold <- FALSE    #Use absolute values of fold-change?
 input_GenePvalCutoff <- 1  #FDR to remove genes 

 input_pathwayMethod = 1  # 1  GAGE
 gagePathwayData.out <- gagePathwayData()  # pathway analysis using GAGE  
   
 results <- gagePathwayData.out  #Enrichment analysis for k-Means clusters  
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction | GAGE analysis: CAX22SHOOTFL vs CAX22SHOOTGC     | statistic | Genes | adj.Pval |
| --------- | ----------------------------------------------- | --------- | ----- | -------- |
| Up        | Plant-type cell wall organization               | 4.1293    | 13    | 3.0e-02  |
|           | Plant-type cell wall organization or biogenesis | 3.3782    | 18    | 3.5e-02  |
|           | Cell wall organization or biogenesis            | 3.1278    | 34    | 3.5e-02  |
|           | External encapsulating structure organization   | 3.0918    | 29    | 3.5e-02  |
|           | Cell wall organization                          | 3.0918    | 29    | 3.5e-02  |

```
 pathwayListData.out = pathwayListData() 
 enrichmentPlot(pathwayListData.out, 25  ) 
```

```
  enrichmentNetwork(pathwayListData.out )  
```

```
  enrichmentNetworkPlotly(pathwayListData.out) 
```

```
 input_pathwayMethod = 3  # 1  fgsea 
 fgseaPathwayData.out <- fgseaPathwayData() #Pathway analysis using fgsea 
```

```
## Warning in fgsea(pathways = gmt, stats = fold, minSize = input_minSetSize, :
## You are trying to run fgseaSimple. It is recommended to use fgseaMultilevel. To
## run fgseaMultilevel, you need to remove the nperm argument in the fgsea function
## call.
```

```
 results <- fgseaPathwayData.out  #Enrichment analysis for k-Means clusters 
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction | GSEA analysis: CAX22SHOOTFL vs CAX22SHOOTGC                               | NES     | Genes | adj.Pval |
| --------- | ------------------------------------------------------------------------- | ------- | ----- | -------- |
| Down      | Biological process involved in interspecies interaction between organisms | -1.8391 | 20    | 1.9e-01  |
| Up        | Plant-type cell wall organization                                         | 1.9587  | 13    | 3.1e-02  |
|           | External encapsulating structure organization                             | 1.8811  | 29    | 3.1e-02  |
|           | Cell wall organization                                                    | 1.8811  | 29    | 3.1e-02  |
|           | Cell wall organization or biogenesis                                      | 1.8406  | 34    | 4.5e-02  |
|           | Plant-type cell wall organization or biogenesis                           | 1.8395  | 18    | 9.1e-02  |

```
  pathwayListData.out = pathwayListData() 
 enrichmentPlot(pathwayListData.out, 25  ) 
```

```
  enrichmentNetwork(pathwayListData.out )  
```

```
  enrichmentNetworkPlotly(pathwayListData.out) 
```

```
   PGSEAplot() # pathway analysis using PGSEA 
```

```
## 
## Computing P values using ANOVA
```

### 8.4 Pathway analysis

```
 input_selectContrast1 <- 'CAX22ROOTFL-CAX22ROOTGC' #select Comparison 
 #input_selectContrast1 = limma.out$comparisons[3] # manually set
 input_selectGO <- 'GOBP'   #Gene set category 
 #input_selectGO='custom' # if custom gmt file
 input_minSetSize <- 5  #Min size for gene set
 input_maxSetSize <- 2000   #Max size for gene set 
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  ) 
 input_pathwayPvalCutoff <- 0.2 #FDR cutoff
 input_nPathwayShow <- 30   #Top pathways to show
 input_absoluteFold <- FALSE    #Use absolute values of fold-change?
 input_GenePvalCutoff <- 1  #FDR to remove genes 

 input_pathwayMethod = 1  # 1  GAGE
 gagePathwayData.out <- gagePathwayData()  # pathway analysis using GAGE  
   
 results <- gagePathwayData.out  #Enrichment analysis for k-Means clusters  
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| “No significant pathway found.” | adj.Pval |
| ------------------------------- | -------- |
| No significant pathway found.   | NULL     |

```
 pathwayListData.out = pathwayListData() 
 enrichmentPlot(pathwayListData.out, 25  ) 
```

```
## NULL
```

```
 input_pathwayMethod = 3  # 1  fgsea 
 fgseaPathwayData.out <- fgseaPathwayData() #Pathway analysis using fgsea 
```

```
## Warning in fgsea(pathways = gmt, stats = fold, minSize = input_minSetSize, :
## You are trying to run fgseaSimple. It is recommended to use fgseaMultilevel. To
## run fgseaMultilevel, you need to remove the nperm argument in the fgsea function
## call.
```

```
 results <- fgseaPathwayData.out  #Enrichment analysis for k-Means clusters 
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| “No significant pathway found.” | adj.Pval |
| ------------------------------- | -------- |
| No significant pathway found.   | NULL     |

```
  pathwayListData.out = pathwayListData() 
 enrichmentPlot(pathwayListData.out, 25  ) 
```

```
## NULL
```

### 8.5 Pathway analysis

```
 input_selectContrast1 <- 'CAX23SHOOTFL-CAX23SHOOTGC'   #select Comparison 
 #input_selectContrast1 = limma.out$comparisons[3] # manually set
 input_selectGO <- 'GOBP'   #Gene set category 
 #input_selectGO='custom' # if custom gmt file
 input_minSetSize <- 5  #Min size for gene set
 input_maxSetSize <- 2000   #Max size for gene set 
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  ) 
 input_pathwayPvalCutoff <- 0.2 #FDR cutoff
 input_nPathwayShow <- 30   #Top pathways to show
 input_absoluteFold <- FALSE    #Use absolute values of fold-change?
 input_GenePvalCutoff <- 1  #FDR to remove genes 

 input_pathwayMethod = 1  # 1  GAGE
 gagePathwayData.out <- gagePathwayData()  # pathway analysis using GAGE  
   
 results <- gagePathwayData.out  #Enrichment analysis for k-Means clusters  
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction | GAGE analysis: CAX23SHOOTFL vs CAX23SHOOTGC       | statistic | Genes | adj.Pval |
| --------- | ------------------------------------------------- | --------- | ----- | -------- |
| Down      | Ribonucleoprotein complex biogenesis              | -3.8024   | 485   | 1.2e-01  |
|           | Protein folding                                   | -3.6642   | 171   | 1.2e-01  |
|           | RNA modification                                  | -3.5849   | 292   | 1.2e-01  |
|           | Ribosome biogenesis                               | -3.5669   | 387   | 1.2e-01  |
|           | NcRNA metabolic process                           | -3.4999   | 490   | 1.2e-01  |
| Up        | Defense response to bacterium                     | 4.2271    | 436   | 2.4e-02  |
|           | Detoxification                                    | 4.0587    | 225   | 2.4e-02  |
|           | Response to toxic substance                       | 4.0552    | 261   | 2.4e-02  |
|           | Response to oomycetes                             | 3.7879    | 105   | 6.1e-02  |
|           | Hydrogen peroxide catabolic process               | 3.5701    | 84    | 7.8e-02  |
|           | Response to salicylic acid                        | 3.5661    | 156   | 7.8e-02  |
|           | Aminoglycan catabolic process                     | 3.5524    | 17    | 9.1e-02  |
|           | Chitin metabolic process                          | 3.5524    | 17    | 9.1e-02  |
|           | Chitin catabolic process                          | 3.5524    | 17    | 9.1e-02  |
|           | Amino sugar catabolic process                     | 3.5524    | 17    | 9.1e-02  |
|           | Glucosamine-containing compound catabolic process | 3.5524    | 17    | 9.1e-02  |
|           | Systemic acquired resistance                      | 3.5394    | 79    | 7.8e-02  |
|           | Hydrogen peroxide metabolic process               | 3.5212    | 104   | 7.8e-02  |
|           | Secondary metabolic process                       | 3.4878    | 322   | 7.8e-02  |
|           | Defense response to oomycetes                     | 3.3265    | 83    | 9.1e-02  |
|           | Toxin metabolic process                           | 3.321     | 55    | 9.1e-02  |
|           | Response to organic cyclic compound               | 3.3022    | 297   | 9.1e-02  |
|           | Response to nutrient levels                       | 3.2411    | 212   | 9.1e-02  |
|           | Cellular oxidant detoxification                   | 3.2185    | 162   | 9.1e-02  |
|           | Cellular detoxification                           | 3.2001    | 176   | 9.1e-02  |
|           | Response to extracellular stimulus                | 3.1507    | 251   | 9.9e-02  |
|           | Cellular response to toxic substance              | 3.1275    | 177   | 1.0e-01  |
|           | Toxin catabolic process                           | 3.0137    | 42    | 1.6e-01  |
|           | Glutathione metabolic process                     | 2.992     | 76    | 1.6e-01  |
|           | Response to starvation                            | 2.9869    | 183   | 1.6e-01  |

```
 pathwayListData.out = pathwayListData() 
 enrichmentPlot(pathwayListData.out, 25  ) 
```

```
  enrichmentNetwork(pathwayListData.out )  
```

```
  enrichmentNetworkPlotly(pathwayListData.out) 
```

```
 input_pathwayMethod = 3  # 1  fgsea 
 fgseaPathwayData.out <- fgseaPathwayData() #Pathway analysis using fgsea 
```

```
## Warning in fgsea(pathways = gmt, stats = fold, minSize = input_minSetSize, :
## You are trying to run fgseaSimple. It is recommended to use fgseaMultilevel. To
## run fgseaMultilevel, you need to remove the nperm argument in the fgsea function
## call.
```

```
 results <- fgseaPathwayData.out  #Enrichment analysis for k-Means clusters 
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction | GSEA analysis: CAX23SHOOTFL vs CAX23SHOOTGC       | NES     | Genes | adj.Pval |
| --------- | ------------------------------------------------- | ------- | ----- | -------- |
| Down      | Starch catabolic process                          | -1.9823 | 17    | 5.1e-02  |
|           | Response to aluminum ion                          | -1.8766 | 14    | 1.1e-01  |
|           | Response to superoxide                            | -1.8453 | 15    | 1.5e-01  |
|           | Response to oxygen radical                        | -1.8453 | 15    | 1.5e-01  |
|           | Removal of superoxide radicals                    | -1.8001 | 13    | 1.6e-01  |
|           | Cellular response to oxygen radical               | -1.8001 | 13    | 1.6e-01  |
|           | Cellular response to superoxide                   | -1.8001 | 13    | 1.6e-01  |
|           | Regulation of cellular response to heat           | -1.7988 | 10    | 1.1e-01  |
|           | Positive regulation of cellular response to heat  | -1.7752 | 6     | 5.6e-02  |
|           | Photosynthetic electron transport chain           | -1.7714 | 52    | 2.6e-02  |
|           | Suspensor development                             | -1.7678 | 7     | 1.0e-01  |
| Up        | Glutathione metabolic process                     | 1.8753  | 76    | 9.2e-03  |
|           | Toxin metabolic process                           | 1.8427  | 55    | 9.3e-03  |
|           | Hydrogen peroxide catabolic process               | 1.8307  | 84    | 9.2e-03  |
|           | Hydrogen peroxide metabolic process               | 1.8163  | 104   | 9.2e-03  |
|           | Systemic acquired resistance                      | 1.8023  | 79    | 9.2e-03  |
|           | Response to oomycetes                             | 1.7976  | 105   | 9.2e-03  |
|           | Cellular response to nitrogen compound            | 1.7943  | 63    | 1.6e-02  |
|           | Aminoglycan metabolic process                     | 1.7896  | 21    | 2.2e-02  |
|           | Activation of innate immune response              | 1.7891  | 41    | 2.0e-02  |
|           | Activation of immune response                     | 1.7891  | 41    | 2.0e-02  |
|           | Response to iron ion starvation                   | 1.778   | 12    | 1.7e-02  |
|           | Detoxification                                    | 1.7763  | 225   | 9.2e-03  |
|           | Aminoglycan catabolic process                     | 1.7756  | 17    | 1.7e-02  |
|           | Chitin metabolic process                          | 1.7756  | 17    | 1.7e-02  |
|           | Chitin catabolic process                          | 1.7756  | 17    | 1.7e-02  |
|           | Amino sugar catabolic process                     | 1.7756  | 17    | 1.7e-02  |
|           | Glucosamine-containing compound catabolic process | 1.7756  | 17    | 1.7e-02  |
|           | Positive regulation of innate immune response     | 1.7731  | 51    | 1.6e-02  |
|           | Toxin catabolic process                           | 1.7702  | 42    | 2.0e-02  |

```
  pathwayListData.out = pathwayListData() 
 enrichmentPlot(pathwayListData.out, 25  ) 
```

```
  enrichmentNetwork(pathwayListData.out )  
```

```
  enrichmentNetworkPlotly(pathwayListData.out) 
```

```
   PGSEAplot() # pathway analysis using PGSEA 
```

```
## 
## Computing P values using ANOVA
```

### 8.6 Pathway analysis

```
 input_selectContrast1 <- 'CAX23ROOTFL-CAX23ROOTGC' #select Comparison 
 #input_selectContrast1 = limma.out$comparisons[3] # manually set
 input_selectGO <- 'GOBP'   #Gene set category 
 #input_selectGO='custom' # if custom gmt file
 input_minSetSize <- 5  #Min size for gene set
 input_maxSetSize <- 2000   #Max size for gene set 
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  ) 
 input_pathwayPvalCutoff <- 0.2 #FDR cutoff
 input_nPathwayShow <- 30   #Top pathways to show
 input_absoluteFold <- FALSE    #Use absolute values of fold-change?
 input_GenePvalCutoff <- 1  #FDR to remove genes 

 input_pathwayMethod = 1  # 1  GAGE
 gagePathwayData.out <- gagePathwayData()  # pathway analysis using GAGE  
   
 results <- gagePathwayData.out  #Enrichment analysis for k-Means clusters  
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction | GAGE analysis: CAX23ROOTFL vs CAX23ROOTGC    | statistic | Genes | adj.Pval |
| --------- | -------------------------------------------- | --------- | ----- | -------- |
| Down      | Photosynthesis                               | -3.8138   | 272   | 1.8e-01  |
| Up        | Cellular response to hypoxia                 | 4.2279    | 233   | 1.1e-02  |
|           | Response to hypoxia                          | 4.1809    | 259   | 1.1e-02  |
|           | Cellular response to decreased oxygen levels | 4.1536    | 235   | 1.1e-02  |
|           | Cellular response to oxygen levels           | 4.1536    | 235   | 1.1e-02  |
|           | Response to oxygen levels                    | 4.0868    | 264   | 1.1e-02  |
|           | Response to decreased oxygen levels          | 4.0816    | 263   | 1.1e-02  |
|           | RNA modification                             | 3.7465    | 297   | 3.6e-02  |
|           | Nuclear division                             | 3.5146    | 208   | 7.3e-02  |
|           | Organelle fission                            | 3.4944    | 249   | 7.3e-02  |
|           | Meiotic nuclear division                     | 3.313     | 117   | 1.3e-01  |
|           | Meiotic cell cycle                           | 3.2627    | 197   | 1.3e-01  |
|           | Meiotic cell cycle process                   | 3.2286    | 162   | 1.3e-01  |
|           | Cell cycle process                           | 3.208     | 465   | 1.3e-01  |

```
 pathwayListData.out = pathwayListData() 
 enrichmentPlot(pathwayListData.out, 25  ) 
```

```
  enrichmentNetwork(pathwayListData.out )  
```

```
  enrichmentNetworkPlotly(pathwayListData.out) 
```

```
 input_pathwayMethod = 3  # 1  fgsea 
 fgseaPathwayData.out <- fgseaPathwayData() #Pathway analysis using fgsea 
```

```
## Warning in fgsea(pathways = gmt, stats = fold, minSize = input_minSetSize, :
## You are trying to run fgseaSimple. It is recommended to use fgseaMultilevel. To
## run fgseaMultilevel, you need to remove the nperm argument in the fgsea function
## call.
```

```
 results <- fgseaPathwayData.out  #Enrichment analysis for k-Means clusters 
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| Direction | GSEA analysis: CAX23ROOTFL vs CAX23ROOTGC         | NES     | Genes | adj.Pval |
| --------- | ------------------------------------------------- | ------- | ----- | -------- |
| Down      | Shoot apical meristem development                 | -1.9396 | 15    | 4.3e-02  |
|           | Regulation of shoot apical meristem development   | -1.8597 | 10    | 4.3e-02  |
|           | Regulation of leaf development                    | -1.8587 | 21    | 6.5e-02  |
|           | Protein-chromophore linkage                       | -1.8215 | 58    | 6.5e-02  |
|           | Photosynthesis, light harvesting in photosystem I | -1.7985 | 23    | 1.1e-01  |
|           | Photosynthesis, light reaction                    | -1.7955 | 143   | 4.3e-02  |
|           | Photosynthesis, light harvesting                  | -1.7931 | 36    | 9.8e-02  |
|           | Photosynthesis                                    | -1.7811 | 272   | 4.3e-02  |
| Up        | Cellular response to hypoxia                      | 1.9936  | 233   | 4.3e-02  |
|           | Cellular response to decreased oxygen levels      | 1.9748  | 235   | 4.3e-02  |
|           | Cellular response to oxygen levels                | 1.9748  | 235   | 4.3e-02  |
|           | Response to hypoxia                               | 1.9555  | 259   | 4.3e-02  |
|           | Response to decreased oxygen levels               | 1.9331  | 263   | 4.3e-02  |
|           | Response to oxygen levels                         | 1.9329  | 264   | 4.3e-02  |
|           | Branched-chain amino acid catabolic process       | 1.9327  | 13    | 4.3e-02  |
|           | Meiotic nuclear division                          | 1.9208  | 117   | 4.3e-02  |
|           | Meiosis I cell cycle process                      | 1.87    | 80    | 4.3e-02  |
|           | Meiosis I                                         | 1.8586  | 74    | 4.3e-02  |
|           | Nuclear chromosome segregation                    | 1.8553  | 122   | 4.3e-02  |
|           | Valine catabolic process                          | 1.8448  | 13    | 1.1e-01  |
|           | Positive regulation of transport                  | 1.8406  | 24    | 1.0e-01  |
|           | Chromosome separation                             | 1.8378  | 41    | 9.8e-02  |
|           | Mitotic cell cycle phase transition               | 1.8143  | 84    | 4.3e-02  |
|           | Reciprocal meiotic recombination                  | 1.8105  | 57    | 6.5e-02  |
|           | Homologous recombination                          | 1.8105  | 57    | 6.5e-02  |
|           | Reciprocal homologous recombination               | 1.8105  | 57    | 6.5e-02  |
|           | Meristem determinacy                              | 1.8095  | 21    | 1.8e-01  |
|           | Cell cycle checkpoint signaling                   | 1.7945  | 43    | 9.8e-02  |
|           | Cell cycle phase transition                       | 1.7933  | 107   | 4.3e-02  |
|           | Regulation of cell cycle phase transition         | 1.7923  | 73    | 6.1e-02  |

```
  pathwayListData.out = pathwayListData() 
 enrichmentPlot(pathwayListData.out, 25  ) 
```

```
  enrichmentNetwork(pathwayListData.out )  
```

```
  enrichmentNetworkPlotly(pathwayListData.out) 
```

```
   PGSEAplot() # pathway analysis using PGSEA 
```

```
## 
## Computing P values using ANOVA
```

### 9.1 Col-0 Chromosome Shoot

```
 input_selectContrast2 <- 'COL00SHOOTFL-COL00SHOOTGC'   #select Comparison 
 #input_selectContrast2 = limma.out$comparisons[3] # manually set
 input_limmaPvalViz <- 0.1  #FDR to filter genes
 input_limmaFCViz <- 2  #FDR to filter genes 
 genomePlotly() # shows fold-changes on the genome 
```

```
## Warning in eval(quote(list(...)), env): NAs introduced by coercion
```

```
## Warning in genomePlotly(): NAs introduced by coercion
```

### 9.2 Col-0 Chromosome Root

```
 input_selectContrast2 <- 'COL00ROOTFL-COL00ROOTGC' #select Comparison 
 #input_selectContrast2 = limma.out$comparisons[3] # manually set
 input_limmaPvalViz <- 0.1  #FDR to filter genes
 input_limmaFCViz <- 2  #FDR to filter genes 
 genomePlotly() # shows fold-changes on the genome 
```

```
## Warning in eval(quote(list(...)), env): NAs introduced by coercion
```

```
## Warning in genomePlotly(): NAs introduced by coercion
```

### 9.3 cax2-2 Chromosome Shoot

```
 input_selectContrast2 <- 'CAX22SHOOTFL-CAX22SHOOTGC'   #select Comparison 
 #input_selectContrast2 = limma.out$comparisons[3] # manually set
 input_limmaPvalViz <- 0.1  #FDR to filter genes
 input_limmaFCViz <- 2  #FDR to filter genes 
 genomePlotly() # shows fold-changes on the genome 
```

```
## Warning in eval(quote(list(...)), env): NAs introduced by coercion
```

```
## Warning in genomePlotly(): NAs introduced by coercion
```

### 9.4 cax2-2 Chromosome Root

```
 input_selectContrast2 <- 'CAX22ROOTFL-CAX22ROOTGC' #select Comparison 
 #input_selectContrast2 = limma.out$comparisons[3] # manually set
 input_limmaPvalViz <- 0.1  #FDR to filter genes
 input_limmaFCViz <- 2  #FDR to filter genes 
 genomePlotly() # shows fold-changes on the genome 
```

```
## Warning in eval(quote(list(...)), env): NAs introduced by coercion
```

```
## Warning in genomePlotly(): NAs introduced by coercion
```

### 9.5 cax2-3 Chromosome Shoot

```
 input_selectContrast2 <- 'CAX23SHOOTFL-CAX23SHOOTGC'   #select Comparison 
 #input_selectContrast2 = limma.out$comparisons[3] # manually set
 input_limmaPvalViz <- 0.1  #FDR to filter genes
 input_limmaFCViz <- 2  #FDR to filter genes 
 genomePlotly() # shows fold-changes on the genome 
```

```
## Warning in eval(quote(list(...)), env): NAs introduced by coercion
```

```
## Warning in genomePlotly(): NAs introduced by coercion
```

### 9.6 cax2-3 Chromosome Root

```
 input_selectContrast2 <- 'CAX23ROOTFL-CAX23ROOTGC' #select Comparison 
 #input_selectContrast2 = limma.out$comparisons[3] # manually set
 input_limmaPvalViz <- 0.1  #FDR to filter genes
 input_limmaFCViz <- 2  #FDR to filter genes 
 genomePlotly() # shows fold-changes on the genome 
```

```
## Warning in eval(quote(list(...)), env): NAs introduced by coercion
```

```
## Warning in genomePlotly(): NAs introduced by coercion
```

### 10. Biclustering 1

```
 input_nGenesBiclust <- 1000    #Top genes for biclustering
 input_biclustMethod <- 'BCCC()'    #Method: 'BCCC', 'QUBIC', 'runibic' ... 
 biclustering.out = biclustering()  # run analysis

 input_selectBicluster <- 1 #select a cluster 
 biclustHeatmap()   # heatmap for selected cluster 
```

```
 input_selectGO4 <- 'GOBP'  #Gene set category 
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO4,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  )  
 results <- geneListBclustGO()  #Enrichment analysis for k-Means clusters   
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| adj.Pval | Genes | Pathways                                       |
| -------- | ----- | ---------------------------------------------- |
| 1.3e-40  | 121   | Anatomical structure morphogenesis             |
| 9.8e-34  | 111   | Cellular localization                          |
| 1.6e-29  | 90    | Intracellular transport                        |
| 3.5e-28  | 91    | Establishment of localization in cell          |
| 2.7e-27  | 141   | Reproduction                                   |
| 8.2e-27  | 130   | Post-embryonic development                     |
| 1.4e-26  | 139   | Reproductive process                           |
| 1.5e-26  | 148   | System development                             |
| 4.1e-24  | 92    | Regulation of biological quality               |
| 1.6e-23  | 119   | Developmental process involved in reproduction |

### 10. Biclustering 2

```
 input_nGenesBiclust <- 1000    #Top genes for biclustering
 input_biclustMethod <- 'BCCC()'    #Method: 'BCCC', 'QUBIC', 'runibic' ... 
 biclustering.out = biclustering()  # run analysis

 input_selectBicluster <- 2 #select a cluster 
 biclustHeatmap()   # heatmap for selected cluster 
```

```
 input_selectGO4 <- 'GOBP'  #Gene set category 
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO4,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  )  
 results <- geneListBclustGO()  #Enrichment analysis for k-Means clusters   
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| adj.Pval | Genes | Pathways                          |
| -------- | ----- | --------------------------------- |
| 7.8e-09  | 8     | Glucose metabolic process         |
| 7.8e-09  | 7     | Carbon fixation                   |
| 7.8e-09  | 13    | Photosynthesis                    |
| 7.8e-09  | 27    | Small molecule metabolic process  |
| 3.2e-08  | 6     | Reductive pentose-phosphate cycle |
| 3.2e-08  | 6     | Photosynthesis, dark reaction     |
| 1.0e-07  | 13    | Carbohydrate biosynthetic process |
| 3.7e-07  | 24    | Response to external stimulus     |
| 9.4e-07  | 19    | Carbohydrate metabolic process    |
| 2.6e-06  | 9     | Monosaccharide metabolic process  |

### 10. Biclustering 3

```
 input_nGenesBiclust <- 1000    #Top genes for biclustering
 input_biclustMethod <- 'BCCC()'    #Method: 'BCCC', 'QUBIC', 'runibic' ... 
 biclustering.out = biclustering()  # run analysis

 input_selectBicluster <- 3 #select a cluster 
 biclustHeatmap()   # heatmap for selected cluster 
```

```
 input_selectGO4 <- 'GOBP'  #Gene set category 
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO4,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  )  
 results <- geneListBclustGO()  #Enrichment analysis for k-Means clusters   
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| adj.Pval | Genes | Pathways                             |
| -------- | ----- | ------------------------------------ |
| 2.6e-03  | 5     | Terpenoid metabolic process          |
| 2.6e-03  | 3     | Triterpenoid metabolic process       |
| 2.6e-03  | 5     | Cellular response to toxic substance |
| 2.6e-03  | 5     | Cellular oxidant detoxification      |
| 2.6e-03  | 5     | Cellular detoxification              |
| 3.7e-03  | 5     | Isoprenoid metabolic process         |
| 6.2e-03  | 5     | Detoxification                       |
| 9.8e-03  | 5     | Response to toxic substance          |
| 9.8e-03  | 9     | Cellular catabolic process           |

### 10. Biclustering 4

```
 input_nGenesBiclust <- 1000    #Top genes for biclustering
 input_biclustMethod <- 'BCCC()'    #Method: 'BCCC', 'QUBIC', 'runibic' ... 
 biclustering.out = biclustering()  # run analysis

 input_selectBicluster <- 4 #select a cluster 
 biclustHeatmap()   # heatmap for selected cluster 
```

```
 input_selectGO4 <- 'GOBP'  #Gene set category 
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO4,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  )  
 results <- geneListBclustGO()  #Enrichment analysis for k-Means clusters   
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| adj.Pval | Genes | Pathways                                                                  |
| -------- | ----- | ------------------------------------------------------------------------- |
| 1.0e-05  | 11    | Response to external stimulus                                             |
| 1.0e-05  | 10    | Response to biotic stimulus                                               |
| 1.0e-05  | 10    | Response to external biotic stimulus                                      |
| 1.0e-05  | 10    | Biological process involved in interspecies interaction between organisms |
| 1.0e-05  | 10    | Response to other organism                                                |
| 1.5e-05  | 9     | Immune system process                                                     |
| 1.5e-05  | 9     | Immune response                                                           |
| 1.5e-05  | 9     | Defense response to other organism                                        |
| 3.0e-04  | 9     | Defense response                                                          |
| 4.3e-04  | 6     | Response to fungus                                                        |

### 11. Co-expression network

```
 input_mySoftPower <- 15    #SoftPower to cutoff
 input_nGenesNetwork <- 1000    #Number of top genes
 input_minModuleSize <- 20  #Module size minimum 
 wgcna.out = wgcna()   # run WGCNA  
```

```
## Warning: executing %dopar% sequentially: no parallel backend registered
```

```
##    Power SFT.R.sq   slope truncated.R.sq mean.k. median.k. max.k.
## 1      1    0.886  1.8400          0.854   550.0     611.0    711
## 2      2    0.556  0.5550          0.555   388.0     438.0    587
## 3      3    0.112  0.1520          0.427   300.0     331.0    511
## 4      4    0.012 -0.0474          0.520   245.0     258.0    458
## 5      5    0.129 -0.1550          0.585   206.0     206.0    418
## 6      6    0.271 -0.2350          0.708   178.0     166.0    386
## 7      7    0.378 -0.2990          0.901   155.0     137.0    360
## 8      8    0.459 -0.3540          0.858   138.0     113.0    337
## 9      9    0.548 -0.3910          0.888   123.0      94.3    317
## 10    10    0.611 -0.4260          0.863   111.0      78.9    300
## 11    12    0.694 -0.4780          0.851    92.0      59.0    271
## 12    14    0.754 -0.5310          0.852    77.8      44.1    247
## 13    16    0.776 -0.5660          0.835    66.7      34.0    227
## 14    18    0.843 -0.5900          0.870    57.9      26.2    210
## 15    20    0.856 -0.6290          0.857    50.8      20.4    195
## TOM calculation: adjacency..
## ..will not use multithreading.
##  Fraction of slow calculations: 0.000000
## ..connectivity..
## ..matrix multiplication (system BLAS)..
## ..normalization..
## ..done.
```

```
 softPower()  # soft power curve 
```

```
  modulePlot()  # plot modules  
```

```
  listWGCNA.Modules.out = listWGCNA.Modules() #modules
```

```
 input_selectGO5 <- 'GOBP'  #Gene set category 
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO5,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  ) 
 input_selectWGCNA.Module <- '1. turquoise (653 genes)' #Select a module
 input_topGenesNetwork <- 20    #SoftPower to cutoff
 input_edgeThreshold <- 0.4 #Number of top genes 
 moduleNetwork()    # show network of top genes in selected module
```

```
##  softConnectivity: FYI: connecitivty of genes with less than 16 valid samples will be returned as NA.
##  ..calculating connectivities..
```

```
 input_removeRedudantSets <- TRUE   #Remove redundant gene sets 
 results <- networkModuleGO()  #Enrichment analysis of selected module
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| adj.Pval | Genes | Pathways                                  |
| -------- | ----- | ----------------------------------------- |
| 7.6e-17  | 59    | Anatomical structure morphogenesis        |
| 1.3e-15  | 60    | Carbohydrate metabolic process            |
| 1.3e-15  | 20    | Movement of cell or subcellular component |
| 1.3e-15  | 18    | Microtubule-based movement                |
| 8.5e-13  | 36    | Cellular carbohydrate metabolic process   |
| 2.8e-12  | 70    | Response to external stimulus             |
| 6.9e-11  | 36    | Ion transmembrane transport               |
| 6.9e-11  | 56    | Transmembrane transport                   |
| 1.9e-10  | 25    | Photosynthesis                            |
| 3.2e-10  | 23    | Microtubule-based process                 |

Second cluster

```
 input_selectGO5 <- 'GOBP'  #Gene set category 
 # Read pathway data again 
 GeneSets.out <-readGeneSets( geneSetFile,
    convertedData.out, input_selectGO5,input_selectOrg,
    c(input_minSetSize, input_maxSetSize)  ) 
## input_selectWGCNA.Module <- '1. turquoise (653 genes)'   #Select a module
 input_selectWGCNA.Module <- '2. blue (341 genes)'  #Select a module
 input_topGenesNetwork <- 20    #SoftPower to cutoff
 input_edgeThreshold <- 0.4 #Number of top genes 
 moduleNetwork()    # show network of top genes in selected module
```

```
##  softConnectivity: FYI: connecitivty of genes with less than 16 valid samples will be returned as NA.
##  ..calculating connectivities..
```

```
 input_removeRedudantSets <- TRUE   #Remove redundant gene sets 
 results <- networkModuleGO()  #Enrichment analysis of selected module
 results$adj.Pval <- format( results$adj.Pval,digits=3 )
 kable( results, row.names=FALSE) 
```

| adj.Pval | Genes | Pathways                              |
| -------- | ----- | ------------------------------------- |
| 2.0e-10  | 35    | RNA processing                        |
| 2.0e-10  | 35    | Cellular localization                 |
| 5.1e-10  | 23    | Ribosome biogenesis                   |
| 1.4e-09  | 29    | Intracellular transport               |
| 4.3e-09  | 24    | Ribonucleoprotein complex biogenesis  |
| 4.5e-09  | 29    | Establishment of localization in cell |
| 9.6e-09  | 18    | RRNA processing                       |
| 1.0e-08  | 18    | RRNA metabolic process                |
| 1.2e-08  | 29    | Protein transport                     |
| 1.2e-08  | 23    | NcRNA metabolic process               |
