---
description: 'TOAST: Test of Arabidopsis Space Transcriptome.'
---

# TOAST: interactive relational database

**TOAST Database Overview**

The TOAST database integrates plant biology and spaceflight omics data within a user-friendly platform powered by the Qlik database management software. Developed to bridge the gap in comparative analyses of spaceflight effects on plant biology, TOAST leverages data from NASA's GeneLab and other repositories, offering comprehensive insights into gene expressions and responses to spaceflight conditions.

**Key Features:**

* **Gene-Centric Visualization:** Facilitates easy exploration of experimental data across various levels, focusing on gene-specific investigations.
* **Multiple Data Sources:** Combines information from prestigious databases, including NASA's GeneLab, enhancing the richness of the dataset.
* **Analytical Tools Integration:** Connects to web-based tools for deeper analyses of spaceflight impacts and related stresses.
* **Advanced Filtering:** Employs gene ontology, Kyoto Encyclopedia of Genes and Genomes pathways, and other filters for detailed genetic network explorations.
* **Insightful Discoveries:** Validates known gene expression patterns and uncovers novel insights, such as mitochondrial function changes in spaceflight conditions.

**Impact and Advancements:**

TOAST addresses the complexities of mining extensive omics data, facilitating intuitive data presentation and analysis. Its development is timely, considering the increasing frequency of plant growth experiments in space, notably on the International Space Station. Through TOAST, researchers can navigate the intricacies of spaceflight-responsive gene expression with greater ease, driving forward our understanding of plant biology in space.

**More Information:**

For further details on TOAST and its applications, visit: \[Astrobiology at the University of Wisconsin-Madison]\(https://astrobiology.botany.wisc.edu/astrobotany

<figure><img src="../.gitbook/assets/image (2) (1) (1) (1) (1) (1).png" alt="" width="549"><figcaption><p>Graphical methods abstract.</p></figcaption></figure>

<figure><img src="../.gitbook/assets/image (28).png" alt=""><figcaption><p><a href="https://www.frontiersin.org/journals/plant-science/articles/10.3389/fpls.2020.00147/full">https://www.frontiersin.org/journals/plant-science/articles/10.3389/fpls.2020.00147/full</a></p></figcaption></figure>

{% embed url="https://gilroy-qlik.botany.wisc.edu/a/sense/app/d97365ed-f15d-48e9-821e-779a7d7b7908/overview" %}
Link to the TOAST application
{% endembed %}

{% embed url="https://docs.google.com/presentation/d/1mxagK6ePXmxyITF0cHbCioN5t6M0itmxMTUcbxMpxKc/edit?usp=sharing" %}
Training slides
{% endembed %}



### The matrix meta-data application identifies similar meta-data features between the most similar plant studies in the OSDR? <a href="#h.g5xbdsdy53rs_l" id="h.g5xbdsdy53rs_l"></a>

{% embed url="https://gilroy-qlik.botany.wisc.edu/a/sense/app/20aa802b-6915-4b1a-87bd-c029a1812e2b/overview" %}
The Matrix is an example of a highlighly connected metadata node found within the OSDR.
{% endembed %}

## &#x20;<a href="#h.p_wlijx2gx56pg_l" id="h.p_wlijx2gx56pg_l"></a>

## Genetic Network Analysis <a href="#h.p_wlijx2gx56pg_l" id="h.p_wlijx2gx56pg_l"></a>

We recommend [STRING](https://string-db.org/cgi/input.pl?sessionId=MPPYe5bLCDPc\&input\_page\_show\_search=off), [SHINY GO](http://bioinformatics.sdstate.edu/go/), [Knetminer](https://knetminer.rothamsted.ac.uk/KnetMiner/) and [for plants the BAR resources t](http://bar.utoronto.ca/)o further statically investigate and visualize the genetic networks identified in TOAST.

**Below:** Example Network diagram produced showing links between spaceflight response genes based on the NCBI literature, protein sequence, co-expression and functional prediction made using gene identified in TOAST and visualized using Knetminer.

<figure><img src="https://lh5.googleusercontent.com/LbKLxCwG2hRojCildMtEuCA0XkjU-kGQdhuWay1Y-DizMkEE8V5PAHZYG_EBBTa5HKtpSyTRcb1EL8aQXHi1Q09FFZ2yWHkNHbMCIlHbMzDMXusP=w1280" alt=""><figcaption></figcaption></figure>

Collaborate with the Gilroy AstroBiology lab by completing this [AstroBotany Class Work Sheet](https://docs.google.com/forms/d/e/1FAIpQLScFqgBl713OUMEoFRxXj\_IiWoyXdsbjaa1j\_XhDaBe\_8Kioxg/viewform?usp=sf\_link). You can even share what you discover so our community can help you interpret the data.  To compare lists of genes found in the different studies we also recommend this app for making [Venn Diagrams](http://jvenn.toulouse.inra.fr/app/example.html).
